<span style="font-size:3em; line-height:36px"><strong>Natural Language Processing with Python
</strong></span>
## Steven Bird, Ewan Klein, and Edward Loper

#### [NOTE] 

이 문서에 나오는 모든 텍스트와 코드는 Steven Bird, Ewan Klein, and Edward Loper의 저서, "Natural Language Processing with Python – Analyzing Text with the Natural Language Toolkit"에서 가져와 일부는 번역하고 일부는 원문 그대로 옮겨 적은 것이다. 또한 내용 중 일부는 교육 목적으로 요약하거나 첨삭하였으나, 원작자의 권리에는 변함이 없음을 알려둔다. 본 자료에 대한 모든 권리는 원작자에게 있음을 알린다.
원래 자료는 https://www.nltk.org/book/ 에서 볼 수 있다.

The following text and code are from the book, "Natural Language Processing with Python – Analyzing Text with the Natural Language Toolkit" by Steven Bird, Ewan Klein, and Edward Loper.
Please note that we have tried to keep the original content as much as possible, and have only modified or translated the content as necessary for teaching purposes.
You can find the original content from: https://www.nltk.org/book/

## Software Requirements

|    |    |
|:----------|:----------------------------------------------------------------------------------|
|**Python:**    |The material presented in this book assumes that you are using Python version 3.2 or later. (Note that NLTK 3.0 also works with Python 2.6 and 2.7.)|
|**NLTK:**|The code examples in this book use NLTK version 3.0. Subsequent releases of NLTK will be backward-compatible with NLTK 3.0.|
|**NLTK-Data:**|This contains the linguistic corpora that are analyzed and processed in the book.|
|**NumPy:**|(recommended) This is a scientific computing library with support for multidimensional arrays and linear algebra, required for certain probability, tagging, clustering, and classification tasks.|
|**Matplotlib:**|(recommended) This is a 2D plotting library for data visualization, and is used in some of the book's code samples that produce line graphs and bar charts.|
|**Stanford NLP Tools:**|(recommended) NLTK includes interfaces to the Stanford NLP Tools which are useful for large scale language processing (see http://nlp.stanford.edu/software/).|
|**NetworkX:**|(optional) This is a library for storing and manipulating network structures consisting of nodes and edges. For visualizing semantic networks, also install the Graphviz library.|
|**Prover9:**|(optional) This is an automated theorem prover for first-order and equational logic, used to support inference in language processing.|

# Natural Language Toolkit (NLTK)

| Language processing task   | NLTK modules           | Functionality |
|:---------------------------|:-----------------------|:--------------|
| Accessing corpora          | corpus                 | standardized interfaces to corpora and lexicons              |
| String processing          | tokenize, stem         | tokenizers, sentence tokenizers, stemmers                    |
| Collocation discovery      | collocations           | t-test, chi-squared, point-wise mutual information           |
| Part-of-speech tagging     | tag                    | n-gram, backoff, Brill, HMM, TnT                             |
| Machine learning           | classify, cluster, tbl | decision tree, maximum entropy, naive Bayes, EM, k-means     |
| Chunking                   | chunk                  | regular expression, n-gram, named-entity                     |
| Parsing                    | parse, ccg             | chart, feature-based, unification, probabilistic, dependency |
| Semantic interpretation    | sem, inference         | lambda calculus, first-order logic, model checking           |
| Evaluation metrics         | metrics                | precision, recall, agreement coefficients                    |
| Probability and estimation | probability            | frequency distributions, smoothed probability distributions  |
| Applications               | app, chat              | graphical concordancer, parsers, WordNet browser, chatbots   |
| Linguistic fieldwork       | toolbox                | manipulate data in SIL Toolbox format                        |

<br>
<br>

<span style="font-size:3em"><strong>Chapter 1. <br><br> Language Processing and Python
</strong></span>
<br>
<br>

---

<span style="font-size:2.5em"><strong>1. Computing with Language: Texts and Words
</strong></span>

# 1.1   Getting Started with Python

In [ ]:
from IPython.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))
%matplotlib inline 

In [1]:
import sys
sys.version_info

sys.version_info(major=3, minor=8, micro=12, releaselevel='final', serial=0)

# 1.2   Getting Started with NLTK

### Jupyter Notebook에서 Package 설치 방법
- 다음과 같이 하지 말것! 단순한 경우에는 제대로 동작하지만 모든 경우에 가능한 것이 아님
- 다음과 같이 할 바에는 차라리 Termoinal에서 사용중인 env로 들어간 이후에 수동으로 설치할 것
```
# conda
!conda install --yes my_package
# pip
!pip install my_package
```

- **다음과 같이 설치할 것:**
```
# conda
!conda install --yes --prefix {sys.prefix} my_package
# pip
!{sys.executable} -m pip install my_package
```

In [2]:
# conda
!conda install --yes --prefix {sys.prefix} nltk

Solving environment: ...working... done

## Package Plan ##

  environment location: C:\Users\82107\.spss\envs\nlp

  added / updated specs:
    - nltk


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2022.3.18  |       haa95532_0         122 KB
    ------------------------------------------------------------
                                           Total:         122 KB

The following packages will be UPDATED:

  ca-certificates                       2022.2.1-haa95532_0 --> 2022.3.18-haa95532_0




ca-certificates-2022 | 122 KB    |            |   0% 
ca-certificates-2022 | 122 KB    | ########## | 100% 
ca-certificates-2022 | 122 KB    | ########## | 100% 
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done


In [3]:
import nltk
nltk.download()

C:\Users\82107\.spss\envs\nlp\lib\site-packages\numpy\__init__.py:148: UserWarning: mkl-service package failed to import, therefore Intel(R) MKL initialization ensuring its correct out-of-the box operation under condition when Gnu OpenMP had already been loaded by Python process is not assured. Please install mkl-service package, see http://github.com/IntelPython/mkl-service
  from . import _distributor_init


showing info https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/index.xml


True

In [4]:
from nltk.book import *

*** Introductory Examples for the NLTK Book ***
Loading text1, ..., text9 and sent1, ..., sent9
Type the name of the text or sentence to view it.
Type: 'texts()' or 'sents()' to list the materials.
text1: Moby Dick by Herman Melville 1851
text2: Sense and Sensibility by Jane Austen 1811
text3: The Book of Genesis
text4: Inaugural Address Corpus
text5: Chat Corpus
text6: Monty Python and the Holy Grail
text7: Wall Street Journal
text8: Personals Corpus
text9: The Man Who Was Thursday by G . K . Chesterton 1908


### nltk.book module
- nltk.book.texts()  [[source](https://www.nltk.org/_modules/nltk/book.html#texts)]
- nltk.book.sents()  [[source](https://www.nltk.org/_modules/nltk/book.html#sents)]

In [5]:
# Book List 
texts()

text1: Moby Dick by Herman Melville 1851
text2: Sense and Sensibility by Jane Austen 1811
text3: The Book of Genesis
text4: Inaugural Address Corpus
text5: Chat Corpus
text6: Monty Python and the Holy Grail
text7: Wall Street Journal
text8: Personals Corpus
text9: The Man Who Was Thursday by G . K . Chesterton 1908


In [6]:
# Sample sentences
sents()

sent1: Call me Ishmael .
sent2: The family of Dashwood had long been settled in Sussex .
sent3: In the beginning God created the heaven and the earth .
sent4: Fellow - Citizens of the Senate and of the House of Representatives :
sent5: I have a problem with people PMing me to lol JOIN
sent6: SCENE 1 : [ wind ] [ clop clop clop ] KING ARTHUR : Whoa there !
sent7: Pierre Vinken , 61 years old , will join the board as a nonexecutive director Nov. 29 .
sent8: 25 SEXY MALE , seeks attrac older single lady , for discreet encounters .
sent9: THE suburb of Saffron Park lay on the sunset side of London , as red and ragged as a cloud of sunset .


`nltk.text.Text` class object

In [7]:
text2

<Text: Sense and Sensibility by Jane Austen 1811>

In [18]:
len(text2)

141576

In [19]:
sent2

['The',
 'family',
 'of',
 'Dashwood',
 'had',
 'long',
 'been',
 'settled',
 'in',
 'Sussex',
 '.']

In [20]:
len(sent2)

11

# 1.3 Searching Text

- `text1.`까지 입력한 후 Tab 키를 눌러 지원되는 메소드를 살펴보자
- `nltk.text.Text` class object에서 지원되는 methods:
>  - concordance(word, width=79, lines=25)
>  - concordance_list(word, width=79, lines=25)
>  - collocation_list(num=20, window_size=2)
>  - collocations(num=20, window_size=2)
>  - count(word)
>  - index(word)
>  - similar(word, num=20)
>  - common_contexts(words, num=20)
>  - ...

In [21]:
text1.concordance("monstrous")

Displaying 11 of 11 matches:
ong the former , one was of a most monstrous size . ... This came towards us , 
ON OF THE PSALMS . " Touching that monstrous bulk of the whale or ork we have r
ll over with a heathenish array of monstrous clubs and spears . Some were thick
d as you gazed , and wondered what monstrous cannibal and savage could ever hav
that has survived the flood ; most monstrous and most mountainous ! That Himmal
they might scout at Moby Dick as a monstrous fable , or still worse and more de
th of Radney .'" CHAPTER 55 Of the Monstrous Pictures of Whales . I shall ere l
ing Scenes . In connexion with the monstrous pictures of whales , I am strongly
ere to enter upon those still more monstrous stories of them which are to be fo
ght have been rummaged out of this monstrous cabinet there is no telling . But 
of Whale - Bones ; for Whales of a monstrous size are oftentimes cast up dead u


concordance를 사용하여 처음 검색하면 인덱싱을 위한 시간이 약간 걸릴 수 있다

**<span style="color:crimson">[Your Turn]</span>** 다른 텍스트에서 다른 단어로 직접 시도해보자.

다음과 같이 시도해 볼 예정이다:  
text2 -> 'affection'  
text3 -> 'lived'  
text4 -> 'nation'  
text5 -> 'lol'

In [22]:
text2.concordance("affection")

Displaying 25 of 79 matches:
, however , and , as a mark of his affection for the three girls , he left them
t . It was very well known that no affection was ever supposed to exist between
deration of politeness or maternal affection on the side of the former , the tw
d the suspicion -- the hope of his affection for me may warrant , without impru
hich forbade the indulgence of his affection . She knew that his mother neither
rd she gave one with still greater affection . Though her late conversation wit
 can never hope to feel or inspire affection again , and if her home be uncomfo
m of the sense , elegance , mutual affection , and domestic comfort of the fami
, and which recommended him to her affection beyond every thing else . His soci
ween the parties might forward the affection of Mr . Willoughby , an equally st
 the most pointed assurance of her affection . Elinor could not be surprised at
he natural consequence of a strong affection in a young and ardent mind . This 
 opinion . 

In [23]:
text3.concordance("lived")

Displaying 25 of 38 matches:
ay when they were created . And Adam lived an hundred and thirty years , and be
ughters : And all the days that Adam lived were nine hundred and thirty yea and
nd thirty yea and he died . And Seth lived an hundred and five years , and bega
ve years , and begat Enos : And Seth lived after he begat Enos eight hundred an
welve years : and he died . And Enos lived ninety years , and begat Cainan : An
 years , and begat Cainan : And Enos lived after he begat Cainan eight hundred 
ive years : and he died . And Cainan lived seventy years and begat Mahalaleel :
rs and begat Mahalaleel : And Cainan lived after he begat Mahalaleel eight hund
years : and he died . And Mahalaleel lived sixty and five years , and begat Jar
s , and begat Jared : And Mahalaleel lived after he begat Jared eight hundred a
and five yea and he died . And Jared lived an hundred sixty and two years , and
o years , and he begat Eno And Jared lived after he begat Enoch eight hundred y
 and two ye

In [24]:
text4.concordance("nation")

Displaying 25 of 330 matches:
 to the character of an independent nation seems to have been distinguished by
f Heaven can never be expected on a nation that disregards the eternal rules o
first , the representatives of this nation , then consisting of little more th
, situation , and relations of this nation and country than any which had ever
, prosperity , and happiness of the nation I have acquired an habitual attachm
an be no spectacle presented by any nation more pleasing , more noble , majest
party for its own ends , not of the nation for the national good . If that sol
tures and the people throughout the nation . On this subject it might become m
if a personal esteem for the French nation , formed in a residence of seven ye
f our fellow - citizens by whatever nation , and if success can not be obtaine
y , continue His blessing upon this nation and its Government and give it all 
powers so justly inspire . A rising nation , spread over a wide and fruitful l
ing now decided by the

In [25]:
text5.concordance("lol")

Displaying 25 of 822 matches:
ast PART 24 / m boo . 26 / m and sexy lol U115 boo . JOIN PART he drew a girl w
ope he didnt draw a penis PART ewwwww lol & a head between her legs JOIN JOIN s
a bowl i got a blunt an a bong ...... lol JOIN well , glad it worked out my cha
e " PART Hi U121 in ny . ACTION would lol @ U121 . . . but appearently she does
30 make sure u buy a nice ring for U6 lol U7 Hi U115 . ACTION isnt falling for 
 didnt ya hear !!!! PART JOIN geeshhh lol U6 PART hes deaf ppl here dont get it
es nobody here i wanna misbeahve with lol JOIN so read it . thanks U7 .. Im hap
ies want to chat can i talk to him !! lol U121 !!! forwards too lol JOIN ALL PE
k to him !! lol U121 !!! forwards too lol JOIN ALL PErvs ... redirect to U121 '
 loves ME the most i love myself JOIN lol U44 how do u know that what ? jerkett
ng wrong ... i can see it in his eyes lol U20 = fiance Jerketts lmao wtf yah I 
cooler by the minute what 'd I miss ? lol noo there too much work ! why not ?? 
 that mean

- `nltk.text.Text.concordance`: 단어 "montrous"는 **... pictures** 와 **... size** 같은 문맥에서 나타나는 것을 봤다. 
- 이와 유사한 범위의 문맥에서는 무슨 단어들이 자주 나올까? `similar` 메소드로 알아볼 수 있다:

In [26]:
text1.similar("monstrous")

true contemptible christian abundant few part mean careful puzzled
mystifying passing curious loving wise doleful gamesome singular
delightfully perilous fearless


In [27]:
text2.similar("monstrous")

very so exceedingly heartily a as good great extremely remarkably
sweet vast amazingly


- "monstrous"은 멜빌과 오스틴에게 다른 느낌으로 사용됨. 오스틴에게는 '긍정'이 함축되어 있고 때로는 "very" 같이 강조 기능을 갖는다.
- `nltk.text.Text.common_contexts`: 둘 또는 그 이상의 단어들에 의해 공유되는 문맥을 조사해볼 수 있게 함.

In [28]:
text2.common_contexts(["monstrous", "very"])

a_pretty am_glad a_lucky is_pretty be_glad


**<span style="color:crimson">[Your Turn]</span>** 다른 단어 쌍을 고르고, 다른 두 개의 텍스트에서 어떻게 사용되는지 살펴보자. `similar()` 와 `common_contexts()` 함수를 사용한다.

- 텍스트 내 단어의 위치를 계산할 수 있다: 텍스트 시작부터 그 단어까지 나오는 단어 수 
- 이 위치 정보를 **dispersion plot**를 사용해서 그릴 수 있다.

#### Dispersion Plot
- 텍스트 내 단어의 위치 정보
- 수직방향 줄: 단어의 출현 사례 한 건에 해당
- 행(row): 전체 텍스트
- text4(Inaugural Address Corpus)에서는 과거 220년간 사용된 단어의 패턴을 볼 수 있다.
- 다른 단어(예: liberty, constitution)를 사용해서 또는 다른 텍스트에서 다른 단어를 사용해 실험해보자.
- 그림을 보기 전에 패턴을 예측할 수 있을까?

In [29]:
text4.dispersion_plot(["citizens", "democracy", "freedom", "duties", "America"])

ValueError: The plot function requires matplotlib to be installed.See http://matplotlib.org/

**Figure 1.2:** Lexical Dispersion Plot for Words in U.S. Presidential Inaugural Addresses: This can be used to investigate changes in language use over time.

In [ ]:
import matplotlib.pyplot as plt
plt.figure(figsize=(20, 8))
text4.dispersion_plot(["citizens", "democracy", "freedom", "duties", "America"])

Now, just for fun, let's try generating some random text in the various styles we have just seen.  
**Note** The generate() method is not available in NLTK 3.0 but will be reinstated in a subsequent version.

**<span style="color:crimson">[Your Turn]</span>** 이제 재미삼아 다른 단어 단어들로 다른 텍스트에서 실험해보자.

### 텍스트 생성
각 텍스트의 다양한 스타일을 따르는 무작위 텍스트를 생성해보자 

`nltk.text.Text.generate(length=100, text_seed=None, random_seed=42)`: Print random text, generated using a <ins>trigram language model</ins>. See also help(`nltk.lm`).

In [30]:
text3.generate()

Building ngram index...


laid by her , and said unto Cain , Where art thou , and said , Go to ,
I will not do it for ten ' s sons ; we dreamed each man according to
their generatio the firstborn said unto Laban , Because I said , Nay ,
but Sarah shall her name be . , duke Elah , duke Shobal , and Akan .
and looked upon my affliction . Bashemath Ishmael ' s blood , but Isra
for as a prince hast thou found of all the cattle in the valley , and
the wo The


"laid by her , and said unto Cain , Where art thou , and said , Go to ,\nI will not do it for ten ' s sons ; we dreamed each man according to\ntheir generatio the firstborn said unto Laban , Because I said , Nay ,\nbut Sarah shall her name be . , duke Elah , duke Shobal , and Akan .\nand looked upon my affliction . Bashemath Ishmael ' s blood , but Isra\nfor as a prince hast thou found of all the cattle in the valley , and\nthe wo The"

In [31]:
text3.generate(text_seed=["looked", "upon", "my"])

Building ngram index...


looked upon my head , and bare a son in his old a and he also had made an end of the
Amorites is not a blessing for me , I will make there at the younge
and the seven thin ears devoured the seven good ears : and Esau came
from Padan , Rachel came with her father ' s cattle . affliction ; now
therefore behold thy wife , Behold , I will make them rulers over my
house . head . affliction . knees , that thy seed be called no more .
head , and dipped the coat in


"looked upon my head , and bare a son in his old a and he also had made an end of the\nAmorites is not a blessing for me , I will make there at the younge\nand the seven thin ears devoured the seven good ears : and Esau came\nfrom Padan , Rachel came with her father ' s cattle . affliction ; now\ntherefore behold thy wife , Behold , I will make them rulers over my\nhouse . head . affliction . knees , that thy seed be called no more .\nhead , and dipped the coat in"

# 1.4 Counting Vocabulary

- 단어를 세는 방법은 여러가지가 있다
- "The Book of Genesis"에 대해서 여러가지 단어 세는 방법을 실습해보자

In [32]:
len(text3)

44764

- 이 숫자는 단어, 구둣점(punctuation symbols) 또는 토큰들을 포함한다.
- 또한 중복도 계산된다
- 고유한 단어(엄격히는 토큰 및 구둣점) 수를 계산하려면?

In [33]:
sorted(set(text3))

['!',
 "'",
 '(',
 ')',
 ',',
 ',)',
 '.',
 '.)',
 ':',
 ';',
 ';)',
 '?',
 '?)',
 'A',
 'Abel',
 'Abelmizraim',
 'Abidah',
 'Abide',
 'Abimael',
 'Abimelech',
 'Abr',
 'Abrah',
 'Abraham',
 'Abram',
 'Accad',
 'Achbor',
 'Adah',
 'Adam',
 'Adbeel',
 'Admah',
 'Adullamite',
 'After',
 'Aholibamah',
 'Ahuzzath',
 'Ajah',
 'Akan',
 'All',
 'Allonbachuth',
 'Almighty',
 'Almodad',
 'Also',
 'Alvah',
 'Alvan',
 'Am',
 'Amal',
 'Amalek',
 'Amalekites',
 'Ammon',
 'Amorite',
 'Amorites',
 'Amraphel',
 'An',
 'Anah',
 'Anamim',
 'And',
 'Aner',
 'Angel',
 'Appoint',
 'Aram',
 'Aran',
 'Ararat',
 'Arbah',
 'Ard',
 'Are',
 'Areli',
 'Arioch',
 'Arise',
 'Arkite',
 'Arodi',
 'Arphaxad',
 'Art',
 'Arvadite',
 'As',
 'Asenath',
 'Ashbel',
 'Asher',
 'Ashkenaz',
 'Ashteroth',
 'Ask',
 'Asshur',
 'Asshurim',
 'Assyr',
 'Assyria',
 'At',
 'Atad',
 'Avith',
 'Baalhanan',
 'Babel',
 'Bashemath',
 'Be',
 'Because',
 'Becher',
 'Bedad',
 'Beeri',
 'Beerlahairoi',
 'Beersheba',
 'Behold',
 'Bela',
 'Belah

In [34]:
len(sorted(set(text3)))

2789

- 이 수는 고유한 단어 2,789개를 의미하며, 이를 2,789개의 **word type**이 있다고 말한다.
- 구둣점 기호는 단어가 아니므로 **type**이라고 지칭하기로 한다.
- 이론 강의시 나온 **types and tokens**의 구분과는 다르다는 것에 주의할 것.

- 텍스트의 lexical richness에 대해서 알아본다.
- 텍스트 전체 단어들에서 단지 6%만 고유한 단어라는 것을 말해준다.
- 각 고유 단어는 평균 16회씩 사용됐다는 것을 말해준다

In [35]:
len(set(text3)) / len(text3)

0.06230453042623537

In [36]:
1/0.06230453042623537

16.050197203298673

- 특정 단어의 출현 횟수는?

In [37]:
text3.count("smote")

5

- 특정 단어가 텍스트 전체에서 차지하는 비중은?

In [38]:
100 * text4.count('a') / len(text4)

1.457806031353621

**<span style="color:crimson">[Your Turn]</span>** text5에서 "lol"의 출현 횟수는? 이 단어가 텍스트 전체에서 차지하는 비중은?

함수로 만들기

In [39]:
def lexical_diversity(text):
    return len(set(text)) / len(text)

def percentage(count, total):
    return 100 * count / total

In [40]:
lexical_diversity(text3)

0.06230453042623537

In [41]:
lexical_diversity(text5)

0.13477005109975562

In [42]:
percentage(text4.count('a'), len(text4))

1.457806031353621

<span style="font-size:2.5em; line-height:36px"><strong>2. A Closer Look at Python: Texts as Lists of Words
</strong></span>

<br><br><br>
## Python 학습에 관한 내용입니다. Python이 아직 익숙하지 않으면 한번 읽어봅시다.
<br><br><br><br><br><br><br><br>

<span style="font-size:2.5em; line-height:36px"><strong>3. Computing with Language: Simple Statistics
</strong></span>

# 3.1   Frequency Distributions

* 문서의 토픽이나 장르에 대한 정보를 가장 잘 담고 있는 단어를 자동으로 찾아낼 수 있을까? 책에서 가장 자주 사용되는 단어 50개를 찾는 문제를 생각해보자.

In [43]:
from nltk.book import *

In [44]:
fdist1 = FreqDist(text1)
fdist1

FreqDist({',': 18713, 'the': 13721, '.': 6862, 'of': 6536, 'and': 6024, 'a': 4569, 'to': 4542, ';': 4072, 'in': 3916, 'that': 2982, ...})

In [45]:
# print() 문을 사용하면 더 많은 정보를 알 수 있다
print(fdist1)

<FreqDist with 19317 samples and 260819 outcomes>


In [46]:
fdist1.most_common(50)

[(',', 18713),
 ('the', 13721),
 ('.', 6862),
 ('of', 6536),
 ('and', 6024),
 ('a', 4569),
 ('to', 4542),
 (';', 4072),
 ('in', 3916),
 ('that', 2982),
 ("'", 2684),
 ('-', 2552),
 ('his', 2459),
 ('it', 2209),
 ('I', 2124),
 ('s', 1739),
 ('is', 1695),
 ('he', 1661),
 ('with', 1659),
 ('was', 1632),
 ('as', 1620),
 ('"', 1478),
 ('all', 1462),
 ('for', 1414),
 ('this', 1280),
 ('!', 1269),
 ('at', 1231),
 ('by', 1137),
 ('but', 1113),
 ('not', 1103),
 ('--', 1070),
 ('him', 1058),
 ('from', 1052),
 ('be', 1030),
 ('on', 1005),
 ('so', 918),
 ('whale', 906),
 ('one', 889),
 ('you', 841),
 ('had', 767),
 ('have', 760),
 ('there', 715),
 ('But', 705),
 ('or', 697),
 ('were', 680),
 ('now', 646),
 ('which', 640),
 ('?', 637),
 ('me', 627),
 ('like', 624)]

### Pretty Printer
위 출력처럼 길게 출력되는 것이 보기 싫고 더 예쁜 출력을 원하면 `pprint.PrettyPrinter`를 사용해보자

In [47]:
import pprint
pp = pprint.PrettyPrinter(compact=True)
pp.pprint(fdist1.most_common(50))

[(',', 18713), ('the', 13721), ('.', 6862), ('of', 6536), ('and', 6024),
 ('a', 4569), ('to', 4542), (';', 4072), ('in', 3916), ('that', 2982),
 ("'", 2684), ('-', 2552), ('his', 2459), ('it', 2209), ('I', 2124),
 ('s', 1739), ('is', 1695), ('he', 1661), ('with', 1659), ('was', 1632),
 ('as', 1620), ('"', 1478), ('all', 1462), ('for', 1414), ('this', 1280),
 ('!', 1269), ('at', 1231), ('by', 1137), ('but', 1113), ('not', 1103),
 ('--', 1070), ('him', 1058), ('from', 1052), ('be', 1030), ('on', 1005),
 ('so', 918), ('whale', 906), ('one', 889), ('you', 841), ('had', 767),
 ('have', 760), ('there', 715), ('But', 705), ('or', 697), ('were', 680),
 ('now', 646), ('which', 640), ('?', 637), ('me', 627), ('like', 624)]


In [48]:
fdist1['whale']

906

**<span style="color:crimson">[Your Turn]</span>** text2에 대해서 frequency distribution을 만들고 빈도 상위 50개를 출력해보자. 단어 "noble"의 빈도를 출력해보자.

In [ ]:
fdist2.most_common

- Mobu Dick 예에서 찾은 단어들 중에서 Moby Dick의 토픽을 짐작하는데 도움이 될만한 단어들이 있었는가?
- "whale"이 빈도 906으로 약간 도움이 될 수도 있겠지만 나머지 단어들은 전혀 도움이 안된다.
- 모두 영어 문장에서 항상 사용되는 단어들 뿐이다.
- 이러한 단어들이 문서에서 차지하는 비율이 얼마나 되는지 알아볼 필요가 있다

### Cumulative Frequency Plot 

In [51]:
pip install numpy

Note: you may need to restart the kernel to use updated packages.


In [49]:
import matplotlib.pyplot as plt
plt.figure(figsize=(20, 8))
fdist1.plot(50, cumulative=True)

ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.8 from "C:\Users\82107\.spss\envs\nlp\python.exe"
  * The NumPy version is: "1.21.5"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: DLL load failed while importing _multiarray_umath: 지정된 모듈을 찾을 수 없습니다.


- 이러한 빈발 단어(frequent words)들이 전체의 50% 이상을 차지한다는 것을 알 수 있다
- 빈발 단어들이 도움이 안된다면 딱 한 번만 사용되는 rare words(소위 hapaxes)는 어떨까?

In [54]:
pp.pprint(fdist1.hapaxes())

['Herman', 'Melville', ']', 'ETYMOLOGY', 'Late', 'Consumptive', 'School',
 'threadbare', 'lexicons', 'mockingly', 'flags', 'mortality', 'signification',
 'HACKLUYT', 'Sw', 'HVAL', 'roundness', 'Dut', 'Ger', 'WALLEN', 'WALW', 'IAN',
 'RICHARDSON', 'KETOS', 'GREEK', 'CETUS', 'LATIN', 'WHOEL', 'ANGLO', 'SAXON',
 'WAL', 'HWAL', 'SWEDISH', 'ICELANDIC', 'BALEINE', 'BALLENA', 'FEGEE',
 'ERROMANGOAN', 'Librarian', 'painstaking', 'burrower', 'grub', 'Vaticans',
 'stalls', 'higgledy', 'piggledy', 'gospel', 'promiscuously', 'commentator',
 'belongest', 'sallow', 'Pale', 'Sherry', 'loves', 'bluntly', 'Subs',
 'thankless', 'Hampton', 'Court', 'hie', 'refugees', 'pampered', 'Michael',
 'Raphael', 'unsplinterable', 'GENESIS', 'JOB', 'JONAH', 'punish', 'ISAIAH',
 'soever', 'cometh', 'incontinently', 'perisheth', 'PLUTARCH', 'MORALS',
 'breedeth', 'Whirlpooles', 'Balaene', 'arpens', 'PLINY', 'Scarcely', 'TOOKE',
 'LUCIAN', 'TRUE', 'catched', 'OCTHER', 'VERBAL', 'TAKEN', 'MOUTH', 'ALFRED',
 '890', 'gudg

 'girlish', 'blossom', 'flowered', '29', 'Enter', 'perpetually', 'Tropic',
 'warmly', 'perfumed', 'sherbet', 'flaked', 'starred', 'conquering', 'Earls',
 'twas', 'seducing', 'witcheries', 'lend', 'spells', 'potencies', 'Inward',
 'twilights', 'texture', 'greybeards', 'oftenest', 'berths', 'cloaked',
 'sentinelled', 'habitually', 'gripping', 'banister', 'abstained', 'patrolling',
 'reverberating', 'crunching', 'regardings', 'unassured', 'deprecating',
 'muffling', 'indistinctly', 'hesitatingly', 'nightly', 'kennel', 'unforseen',
 'excitedly', 'temptation', 'emboldened', 'begone', 'Anyway', 'coverlid', 'Tic',
 'Dolly', 'appointments', 'snooze', 'eleventh', 'commandment', 'twelfth',
 'blazes', 'jackasses', 'Maybe', 'afoul', 'plaguey', 'Pipe', 'Lighting',
 'Norse', 'thrones', 'fabricated', 'tradition', 'tripod', 'symbolized', 'Khan',
 'soothes', 'pleasuring', 'whiffs', 'sereneness', 'lurchingly', '31', 'Mab',
 'stubbing', 'confoundedly', 'cudgelling', 'whaleboning', 'farmer', 'whittled',
 

 'spangling', 'opal', 'Gentlemen', 'pervades', 'mapped', 'slacken',
 'whitenesses', 'keeling', 'resided', 'foremastmen', 'underwent', 'poop',
 'reinforcement', 'corals', 'assuring', 'Adios', 'Watching', 'beached',
 'befriended', 'providentially', 'Tahitians', 'Chartering', 'refuses',
 'convictions', 'advised', 'Auto', 'da', 'Fe', 'archiepiscopacy', 'withdraw',
 'Excuse', 'procuring', '55', 'Monstrous', 'advert', 'portraits', 'challenge',
 'proving', 'pictorial', 'delusions', 'inventive', 'unscrupulous', 'panellings',
 'pedestals', 'statues', 'shields', 'medallions', 'Saladin', 'cavern',
 'Brahmins', 'prefigured', 'occurs', 'Vishnu', 'Matse', 'Avatar', 'section',
 'Galleries', 'succeeds', 'rescuing', 'Hogarth', 'Descending', 'Hogarthian',
 'howdah', 'Traitors', 'Gate', 'Prodromus', 'primers', 'imitated', 'vases',
 'publisher', 'Revival', 'vignettes', 'embellishments', 'eau', 'unexhausted',
 'Advancement', 'unprofessional', 'Peterson', 'rafts', 'Fisheries', 'Picture',
 '1793', 'veracious

 'allaying', 'bathing', 'co', 'sentimentally', 'acerbities', 'envy',
 'attainable', 'saddle', 'paradise', 'thicker', 'oblongs', 'Berkshire', 'Plum',
 'participating', 'refreshing', 'tint', 'bestreaked', 'plums', 'rubies',
 'citron', 'cutlet', 'le', 'Gros', 'vintage', 'vineyards', 'Champagne',
 'slobgollion', 'ineffably', 'stringy', 'ruptured', 'membranes', 'coalescing',
 'Gurry', 'designates', 'glutinous', 'Nippers', 'indigenous', 'vocabulary',
 'nipper', 'Edgewise', 'operates', 'squilgee', 'impurities', 'receptacle',
 'arrives', 'tyros', 'workmen', 'lurches', 'hone', 'spademan', 'shoeless',
 'assistants', 'Toes', '95', 'Cassock', 'mortemizing', 'strolled', 'scanned',
 'scuppers', 'cone', 'Maachah', 'Judea', 'Asa', 'depose', 'abomination',
 'Kedron', 'grandissimus', 'Extending', 'cylindrically', 'boa', 'pantaloon',
 'slits', 'canonicals', 'Immemorial', 'minced', 'orator', 'Arrayed',
 'occupying', 'bible', 'archbishopric', '!*', 'enjoins', 'accelerated',
 'improving', '96', 'Works', 'ou

In [55]:
len(fdist1.hapaxes())

9002

- hapaxes가 9000여개나 된다! 너무 많아서 도움이 않된다!
- 다른 방법이 필요하다!!

# 3.2   Fine-grained Selection of Words

- 이번에는 긴 단어를 보자. 아마도 더 도움이 될지도 모른다.
- 길이가 15글자보다 긴 단어들의 집합을 다음과 같이 정의한다.  
    <span style="font-size:1.5em; line-height:36px">𝑃(𝑤)={𝑤|len(𝑤)>15}</span>
- 조사 대상 문서의 어휘를 _V_ 라고 하면, 길이 15 이상인 단어의 집합 _S_ 는  
    <span style="font-size:1.5em; line-height:36px">𝑆={𝑤│𝑤∈𝑉 𝑎𝑛𝑑 𝑤∈𝑃(𝑤)}</span>


In [56]:
V = set(text1)
long_words = [w for w in V if len(w) > 15]  # [주의] 리스트이므로 중복될 수 있다
sorted(long_words)

['CIRCUMNAVIGATION',
 'Physiognomically',
 'apprehensiveness',
 'cannibalistically',
 'characteristically',
 'circumnavigating',
 'circumnavigation',
 'circumnavigations',
 'comprehensiveness',
 'hermaphroditical',
 'indiscriminately',
 'indispensableness',
 'irresistibleness',
 'physiognomically',
 'preternaturalness',
 'responsibilities',
 'simultaneousness',
 'subterraneousness',
 'supernaturalness',
 'superstitiousness',
 'uncomfortableness',
 'uncompromisedness',
 'undiscriminating',
 'uninterpenetratingly']

**<span style="color:crimson">[Your Turn]</span>** 다른 text에 대해서 실험해보고, 또, 길이 조건을 바꿔서 실험해보자.

- text4의 long words 목록에는 "constitutionally", "transcontinental" 같은 단어들이 있어서 국가 관련된 것이라는 정보를 준다. - text5에서는 "boooooooooooglyyyyyy" 와 "yuuuuuuuuuuuummmmmmmmmmmm" 같은 단어들이 있어서 내용이 비형식적이며 잡담 같은 것이라는 정보를 준다.
- 이제 토픽이나 장르에 대한 정보를 가장 잘 담고 있는 단어를 자동으로 찾아낸 것인가?
- 짧은 빈발 단어 보다는 긴 단어가 좋기는 하지만 지금 찾은 것은 hapaxes 일 가능성이 높다. 따라서 단어가 적당히 길고 출현 횟수도 적당히 있는 단어들이 도움이 될 것 같다.
- 길이가 7보다 크고 빈도가 7번보다 큰 것들을 찾아보자

In [57]:
fdist5 = FreqDist(text5)
sorted(w for w in set(text5) if len(w) > 7 and fdist5[w] > 7)

['#14-19teens',
 '#talkcity_adults',
 '((((((((((',
 '........',
 'Question',
 'actually',
 'anything',
 'computer',
 'cute.-ass',
 'everyone',
 'football',
 'innocent',
 'listening',
 'remember',
 'seriously',
 'something',
 'together',
 'tomorrow',
 'watching']

드디어 자주 출현하고 내용을 담은 단어들을 찾은 것 같다(아직 많이 부족하지만). 

# 3.3   Collocations and Bigrams

- **Collocation(연어)** : 유난히 함께 출현하는 단어열.  
  - collocation의 예: 'red wine', 'bright idea', 'talk freely'
  - collocation이 아닌 예 : 'the wine', 'this idea'   
- collocation의 특성: Native speaker에게 자연스럽게 보이는 조합이다. 따라서 어떤 한 단어를 비슷한 뜻의 단어로 치환하면 뜻이 매우 부자연스럽게 느껴진다.

|natural English...	|unnatural English... |
|-------------------|:-------------------:|
|the fast train     |the ~~quick~~ train|
|fast food          |~~quick~~ food|
|a quick shower     |a ~~fast~~ shower|
|a quick meal       |a ~~fast~~ meal|   

- Collocation은 **frequent bigram**이므로 collocation 을 처리하기 위해서는 우선 텍스트에서 단어 쌍(즉, **bigram**)들을 추출해야 한다.

In [58]:
list(bigrams(['more', 'is', 'said', 'than', 'done']))

[('more', 'is'), ('is', 'said'), ('said', 'than'), ('than', 'done')]

In [59]:
bigrams(['more', 'is', 'said', 'than', 'done'])

<generator object bigrams at 0x0000025BCC28C9E0>

### NLTK의 Collocation 추출 함수
`nltk.text.Text.collocations(num=20, window_size=2)`
- num (int) : The maximum number of collocations to print.
- window_size (int) : The number of tokens spanned by a collocation (default=2)

In [60]:
text4.collocations()

ValueError: too many values to unpack (expected 2)

In [61]:
text8.collocations()

ValueError: too many values to unpack (expected 2)

# 3.4   Counting Other Things
### The distribution of word lengths in a text

In [62]:
pp.pprint([len(w) for w in text1])

[1, 4, 4, 2, 6, 8, 4, 1, 9, 1, 1, 8, 2, 1, 4, 11, 5, 2, 1, 7, 6, 1, 3, 4, 5, 2,
 10, 2, 4, 1, 5, 1, 4, 1, 3, 5, 1, 1, 3, 3, 3, 1, 2, 3, 4, 7, 3, 3, 8, 3, 8, 1,
 4, 1, 5, 12, 1, 9, 11, 4, 3, 3, 3, 5, 2, 3, 3, 5, 7, 2, 3, 5, 1, 2, 5, 2, 4, 3,
 3, 8, 1, 2, 7, 6, 8, 3, 2, 3, 9, 1, 1, 5, 3, 4, 2, 4, 2, 6, 6, 1, 3, 2, 5, 4,
 2, 4, 4, 1, 5, 1, 4, 2, 2, 2, 6, 2, 3, 6, 7, 3, 1, 7, 9, 1, 3, 6, 1, 1, 5, 6,
 5, 6, 3, 13, 2, 3, 4, 1, 3, 7, 4, 5, 2, 3, 4, 2, 2, 8, 1, 5, 1, 3, 2, 1, 3, 3,
 1, 4, 1, 4, 6, 2, 5, 4, 9, 2, 7, 1, 3, 2, 3, 1, 5, 2, 6, 2, 7, 2, 2, 7, 1, 1,
 10, 1, 5, 1, 3, 2, 2, 4, 11, 4, 3, 3, 1, 3, 3, 1, 6, 1, 1, 1, 1, 1, 4, 1, 3, 1,
 2, 4, 1, 2, 6, 2, 2, 10, 1, 1, 10, 5, 1, 5, 1, 5, 1, 5, 1, 5, 1, 5, 1, 5, 1, 5,
 1, 6, 1, 3, 1, 5, 1, 4, 1, 7, 1, 5, 1, 9, 1, 5, 1, 7, 1, 7, 1, 6, 1, 7, 1, 7,
 1, 5, 1, 4, 1, 4, 1, 5, 1, 5, 1, 4, 1, 4, 1, 11, 1, 8, 1, 8, 2, 1, 3, 1, 3, 1,
 9, 2, 2, 4, 2, 4, 4, 4, 4, 11, 8, 3, 4, 1, 4, 2, 1, 4, 5, 2, 1, 3, 1, 3, 7, 2,
 4, 4, 7, 3, 4, 8, 3, 6, 1, 6, 2, 3, 5, 1

 1, 2, 4, 7, 3, 4, 1, 2, 3, 3, 9, 4, 2, 1, 5, 5, 5, 5, 2, 3, 4, 2, 3, 4, 1, 11,
 2, 4, 7, 11, 4, 5, 3, 1, 3, 4, 5, 3, 2, 7, 3, 5, 6, 1, 1, 2, 2, 6, 1, 2, 4, 1,
 3, 4, 6, 5, 4, 1, 5, 3, 4, 10, 2, 8, 3, 5, 2, 2, 9, 4, 1, 1, 2, 1, 3, 4, 3, 5,
 1, 3, 5, 3, 4, 2, 3, 5, 4, 4, 2, 1, 5, 2, 5, 1, 5, 5, 2, 5, 1, 3, 4, 2, 3, 9,
 3, 2, 2, 5, 4, 1, 7, 1, 8, 1, 4, 2, 7, 1, 4, 5, 3, 2, 5, 4, 3, 4, 9, 2, 6, 6,
 2, 2, 5, 2, 2, 5, 1, 3, 7, 10, 2, 8, 2, 2, 5, 1, 1, 7, 1, 5, 4, 5, 1, 6, 3, 3,
 3, 6, 4, 2, 3, 5, 1, 2, 2, 1, 3, 1, 4, 2, 7, 3, 3, 6, 3, 10, 3, 11, 1, 8, 1, 4,
 6, 7, 4, 5, 3, 5, 1, 8, 2, 2, 1, 4, 1, 7, 8, 2, 2, 4, 1, 8, 1, 4, 6, 13, 7, 6,
 6, 10, 1, 3, 8, 2, 3, 4, 2, 5, 7, 1, 4, 1, 3, 10, 8, 2, 5, 3, 4, 2, 5, 4, 2, 2,
 1, 4, 2, 8, 1, 6, 5, 9, 2, 7, 2, 4, 12, 8, 4, 3, 6, 1, 3, 12, 8, 6, 1, 1, 4, 3,
 2, 4, 1, 1, 7, 2, 4, 4, 2, 3, 2, 3, 2, 4, 2, 1, 3, 1, 4, 2, 2, 10, 3, 6, 3, 4,
 1, 4, 1, 13, 8, 4, 6, 7, 4, 3, 5, 1, 1, 7, 4, 2, 3, 4, 1, 5, 2, 7, 10, 2, 4, 1,
 2, 4, 3, 4, 2, 1, 6, 3, 3, 2, 5, 6, 1

 3, 8, 10, 1, 3, 10, 2, 8, 4, 3, 6, 6, 7, 3, 4, 2, 3, 5, 1, 3, 2, 4, 4, 2, 3,
 10, 4, 4, 2, 4, 1, 7, 4, 1, 5, 3, 2, 10, 11, 1, 3, 4, 4, 7, 3, 10, 3, 3, 1, 4,
 1, 6, 1, 10, 1, 5, 4, 2, 9, 8, 2, 3, 6, 2, 3, 7, 4, 5, 4, 1, 3, 1, 13, 5, 8, 2,
 1, 8, 5, 1, 1, 5, 1, 5, 1, 8, 7, 5, 1, 6, 2, 5, 1, 7, 3, 10, 1, 3, 3, 5, 1, 4,
 2, 10, 1, 4, 1, 8, 1, 12, 9, 5, 2, 4, 6, 5, 3, 2, 2, 1, 4, 3, 13, 4, 2, 4, 4,
 8, 2, 4, 3, 4, 4, 10, 8, 5, 1, 4, 3, 4, 1, 6, 1, 3, 1, 4, 1, 9, 4, 5, 4, 3, 7,
 3, 2, 1, 1, 3, 5, 3, 2, 1, 8, 4, 3, 2, 1, 1, 3, 9, 6, 2, 3, 4, 6, 8, 3, 2, 1,
 1, 1, 7, 5, 3, 2, 1, 1, 1, 11, 6, 5, 3, 2, 1, 1, 3, 8, 1, 2, 2, 3, 8, 6, 2, 4,
 1, 3, 2, 4, 3, 5, 7, 7, 2, 4, 3, 10, 9, 2, 3, 7, 1, 1, 5, 1, 4, 4, 5, 3, 1, 3,
 3, 3, 4, 4, 5, 1, 3, 4, 1, 4, 2, 3, 4, 1, 5, 11, 2, 1, 8, 4, 1, 4, 3, 5, 9, 7,
 1, 2, 4, 1, 3, 6, 1, 1, 6, 6, 4, 1, 1, 5, 6, 2, 2, 3, 1, 6, 5, 4, 3, 10, 8, 2,
 4, 4, 7, 4, 4, 1, 9, 4, 3, 7, 1, 3, 7, 10, 1, 4, 1, 6, 2, 1, 5, 9, 1, 3, 4, 1,
 9, 4, 9, 5, 4, 3, 5, 10, 5, 5, 7, 1, 3, 2,

 4, 2, 1, 5, 1, 3, 10, 3, 1, 3, 4, 2, 2, 1, 4, 4, 3, 7, 1, 3, 2, 4, 6, 2, 7, 2,
 4, 3, 4, 2, 7, 3, 5, 1, 4, 1, 7, 3, 4, 5, 3, 2, 8, 1, 8, 2, 3, 1, 5, 5, 7, 2,
 3, 6, 1, 4, 4, 6, 2, 4, 4, 2, 5, 1, 2, 5, 1, 4, 3, 4, 2, 4, 2, 4, 1, 3, 4, 2,
 7, 2, 3, 5, 8, 2, 2, 1, 1, 10, 1, 5, 2, 1, 5, 3, 2, 1, 8, 3, 2, 3, 1, 7, 5, 3,
 9, 1, 3, 4, 8, 2, 4, 1, 1, 9, 4, 4, 10, 1, 2, 3, 6, 2, 3, 7, 7, 1, 4, 4, 3, 4,
 1, 7, 9, 1, 3, 4, 2, 2, 1, 7, 1, 1, 5, 3, 1, 2, 1, 1, 4, 3, 7, 1, 1, 3, 3, 2,
 6, 2, 3, 4, 2, 4, 1, 3, 4, 1, 4, 2, 4, 2, 3, 9, 10, 1, 4, 4, 2, 2, 1, 1, 4, 1,
 5, 5, 5, 1, 3, 10, 11, 2, 3, 7, 2, 9, 1, 2, 2, 4, 1, 2, 5, 2, 7, 3, 1, 4, 4, 2,
 8, 7, 1, 3, 1, 5, 5, 2, 1, 3, 3, 1, 1, 7, 1, 5, 3, 4, 1, 8, 6, 3, 1, 7, 1, 1,
 3, 5, 4, 2, 3, 5, 4, 1, 3, 7, 3, 3, 5, 8, 5, 13, 7, 4, 3, 4, 1, 3, 1, 5, 3, 5,
 5, 4, 7, 7, 2, 4, 9, 1, 4, 10, 5, 7, 2, 2, 3, 1, 3, 1, 5, 4, 10, 6, 6, 3, 3, 1,
 2, 9, 8, 2, 2, 1, 3, 9, 6, 3, 1, 4, 2, 8, 1, 3, 1, 4, 1, 4, 6, 4, 3, 4, 2, 1,
 7, 5, 2, 3, 3, 5, 2, 3, 6, 2, 3, 4, 1, 2, 

 4, 1, 7, 5, 4, 4, 5, 2, 4, 3, 4, 2, 4, 1, 13, 10, 1, 3, 5, 3, 5, 5, 1, 7, 1, 3,
 6, 3, 3, 7, 6, 2, 4, 4, 8, 3, 7, 1, 2, 10, 3, 3, 1, 6, 1, 2, 6, 4, 1, 7, 2, 3,
 10, 1, 6, 2, 3, 6, 8, 1, 2, 1, 4, 3, 1, 4, 1, 3, 8, 4, 5, 5, 6, 2, 3, 5, 7, 4,
 1, 4, 4, 3, 6, 1, 6, 1, 7, 1, 3, 3, 1, 4, 3, 6, 2, 3, 7, 1, 3, 5, 3, 4, 4, 6,
 4, 3, 4, 11, 1, 9, 1, 3, 8, 2, 4, 3, 8, 1, 3, 2, 3, 1, 5, 3, 7, 2, 1, 5, 5, 1,
 3, 2, 3, 4, 3, 2, 8, 1, 4, 5, 2, 4, 5, 4, 3, 7, 4, 4, 2, 2, 7, 9, 2, 3, 5, 2,
 8, 1, 2, 2, 2, 1, 5, 2, 3, 4, 7, 3, 6, 2, 8, 3, 1, 4, 4, 2, 4, 1, 3, 4, 6, 2,
 7, 3, 7, 5, 2, 4, 2, 1, 2, 3, 3, 7, 1, 2, 2, 1, 4, 2, 3, 1, 4, 6, 1, 3, 3, 4,
 6, 1, 1, 4, 1, 4, 1, 2, 4, 3, 4, 1, 3, 7, 2, 3, 3, 4, 4, 1, 3, 2, 3, 6, 1, 4,
 2, 4, 4, 4, 4, 5, 4, 1, 3, 1, 2, 5, 2, 4, 1, 7, 2, 3, 7, 4, 3, 4, 4, 9, 1, 4,
 6, 1, 5, 3, 7, 4, 7, 1, 4, 2, 3, 7, 1, 6, 4, 4, 1, 3, 7, 4, 4, 4, 7, 6, 2, 1,
 7, 1, 2, 3, 4, 4, 3, 4, 12, 8, 5, 6, 5, 7, 1, 3, 4, 8, 4, 2, 8, 1, 3, 1, 3, 5,
 5, 6, 3, 7, 7, 4, 4, 3, 8, 1, 7, 1, 3, 6, 6, 

 8, 2, 3, 1, 7, 3, 9, 8, 5, 1, 3, 7, 6, 5, 2, 7, 1, 3, 7, 9, 5, 3, 5, 2, 8, 1,
 5, 5, 5, 7, 2, 2, 6, 6, 1, 3, 7, 3, 5, 2, 2, 2, 1, 5, 6, 7, 4, 3, 5, 1, 3, 5,
 3, 6, 5, 1, 4, 7, 6, 7, 2, 3, 8, 1, 3, 3, 2, 5, 1, 3, 5, 5, 1, 4, 1, 2, 4, 10,
 4, 1, 4, 1, 3, 1, 7, 3, 4, 1, 1, 5, 1, 4, 1, 5, 5, 3, 9, 12, 2, 4, 8, 4, 1, 4,
 3, 5, 5, 4, 3, 2, 3, 8, 2, 3, 6, 1, 5, 1, 1, 7, 2, 3, 4, 7, 3, 4, 1, 1, 8, 4,
 8, 10, 2, 5, 1, 1, 4, 1, 3, 3, 4, 1, 7, 4, 7, 3, 5, 4, 3, 6, 2, 3, 4, 5, 8, 1,
 3, 4, 1, 5, 3, 3, 1, 6, 2, 6, 6, 1, 5, 9, 1, 9, 9, 4, 9, 2, 3, 4, 1, 6, 1, 2,
 5, 1, 10, 8, 6, 1, 2, 3, 4, 7, 3, 5, 1, 5, 6, 5, 5, 2, 4, 1, 3, 4, 6, 3, 9, 5,
 1, 2, 5, 2, 3, 5, 3, 9, 4, 4, 5, 3, 5, 1, 3, 4, 4, 3, 10, 8, 5, 2, 6, 3, 3, 8,
 6, 1, 3, 4, 13, 2, 3, 4, 4, 3, 4, 6, 3, 1, 3, 5, 1, 3, 7, 1, 3, 3, 4, 1, 3, 3,
 4, 1, 1, 2, 1, 2, 2, 10, 5, 2, 2, 2, 2, 6, 1, 1, 8, 7, 1, 2, 2, 5, 1, 3, 3, 8,
 2, 2, 4, 3, 3, 2, 11, 2, 1, 4, 3, 3, 5, 1, 5, 2, 7, 7, 4, 2, 3, 3, 1, 5, 7, 1,
 3, 4, 10, 3, 8, 3, 1, 2, 3, 9, 2, 3, 5, 4, 

 3, 5, 2, 2, 6, 7, 1, 2, 3, 4, 7, 2, 4, 9, 6, 3, 4, 4, 1, 4, 3, 2, 4, 1, 7, 1,
 5, 4, 2, 4, 1, 3, 4, 7, 6, 1, 7, 2, 3, 2, 2, 5, 1, 3, 7, 6, 2, 5, 1, 1, 4, 1,
 3, 2, 6, 7, 2, 4, 4, 1, 2, 5, 3, 4, 2, 11, 3, 8, 2, 5, 2, 3, 1, 7, 2, 3, 3, 4,
 4, 3, 3, 8, 6, 6, 3, 4, 2, 6, 1, 2, 2, 8, 4, 6, 3, 6, 7, 2, 3, 8, 1, 6, 5, 2,
 4, 9, 2, 3, 6, 4, 1, 4, 7, 2, 4, 4, 1, 3, 6, 3, 2, 2, 3, 8, 3, 4, 1, 7, 8, 4,
 3, 7, 3, 6, 5, 5, 2, 3, 13, 6, 1, 1, 1, 5, 1, 2, 1, 5, 1, 1, 11, 1, 12, 9, 1,
 3, 3, 1, 2, 3, 6, 2, 3, 4, 4, 8, 4, 7, 2, 4, 4, 2, 6, 1, 5, 1, 6, 4, 2, 4, 2,
 4, 5, 1, 3, 7, 2, 4, 1, 6, 6, 2, 4, 1, 3, 5, 5, 4, 3, 8, 1, 2, 2, 4, 1, 4, 6,
 1, 3, 4, 4, 1, 6, 10, 2, 3, 7, 2, 3, 3, 3, 5, 6, 1, 3, 3, 3, 7, 3, 5, 10, 4, 4,
 4, 4, 5, 2, 4, 4, 1, 3, 3, 1, 5, 7, 6, 4, 2, 4, 8, 7, 2, 2, 1, 4, 2, 3, 1, 7,
 2, 6, 2, 2, 4, 2, 4, 6, 9, 3, 4, 1, 1, 2, 4, 4, 6, 9, 3, 3, 8, 1, 1, 6, 2, 9,
 1, 1, 3, 4, 5, 2, 3, 9, 12, 15, 2, 7, 1, 4, 3, 1, 7, 4, 1, 4, 6, 1, 4, 3, 6, 7,
 5, 5, 3, 9, 1, 2, 3, 6, 3, 8, 4, 3, 2, 3, 5, 1

 3, 1, 2, 5, 4, 3, 1, 6, 3, 1, 3, 1, 3, 8, 2, 3, 7, 1, 3, 3, 11, 2, 2, 3, 6, 5,
 6, 2, 3, 3, 4, 1, 4, 2, 8, 2, 2, 3, 10, 2, 3, 2, 3, 4, 4, 6, 2, 3, 9, 1, 3, 8,
 2, 3, 7, 2, 4, 6, 5, 1, 2, 2, 6, 3, 1, 3, 6, 3, 3, 8, 1, 2, 5, 1, 2, 7, 6, 4,
 2, 5, 3, 8, 2, 6, 4, 3, 3, 1, 4, 2, 3, 3, 4, 1, 3, 3, 10, 2, 3, 5, 2, 5, 7, 1,
 6, 9, 2, 3, 9, 4, 4, 2, 6, 1, 5, 6, 2, 3, 8, 1, 3, 4, 7, 4, 2, 3, 8, 4, 4, 2,
 4, 1, 6, 5, 6, 2, 3, 9, 1, 3, 4, 4, 1, 4, 3, 3, 5, 3, 2, 3, 5, 3, 5, 3, 1, 5,
 7, 10, 2, 3, 4, 4, 7, 2, 2, 5, 1, 10, 2, 1, 2, 3, 6, 1, 8, 8, 4, 3, 6, 9, 2, 3,
 5, 5, 2, 9, 2, 4, 2, 4, 2, 3, 8, 4, 1, 7, 1, 3, 10, 5, 6, 2, 3, 2, 3, 2, 3, 9,
 1, 7, 1, 2, 4, 2, 7, 5, 1, 6, 2, 3, 4, 1, 3, 3, 3, 4, 8, 2, 1, 9, 10, 2, 7, 3,
 3, 1, 2, 2, 4, 4, 2, 9, 5, 5, 3, 2, 9, 1, 3, 8, 6, 4, 7, 5, 1, 3, 9, 2, 5, 1,
 4, 1, 5, 4, 3, 5, 1, 5, 2, 2, 3, 3, 1, 4, 1, 7, 2, 5, 2, 2, 3, 7, 1, 3, 5, 2,
 3, 5, 1, 5, 4, 5, 3, 2, 3, 5, 4, 1, 2, 4, 4, 3, 3, 1, 4, 6, 3, 1, 6, 4, 1, 7,
 1, 7, 1, 3, 4, 9, 2, 4, 11, 2, 3, 4, 1, 3, 1

 3, 7, 8, 1, 3, 4, 1, 4, 8, 4, 1, 4, 6, 3, 7, 6, 2, 3, 8, 1, 4, 9, 3, 4, 3, 1,
 3, 1, 5, 3, 1, 10, 3, 1, 6, 1, 2, 3, 9, 6, 4, 1, 7, 1, 2, 3, 3, 5, 1, 2, 5, 4,
 2, 5, 1, 6, 1, 2, 3, 3, 1, 4, 4, 1, 3, 7, 2, 3, 2, 10, 8, 2, 5, 1, 11, 4, 4, 3,
 2, 4, 1, 4, 6, 3, 1, 5, 1, 4, 1, 2, 4, 3, 4, 1, 8, 3, 8, 7, 2, 3, 1, 4, 3, 4,
 10, 7, 1, 4, 3, 5, 6, 9, 2, 1, 6, 2, 1, 7, 1, 5, 1, 3, 2, 2, 4, 4, 3, 1, 2, 9,
 2, 5, 1, 5, 4, 4, 1, 9, 3, 8, 8, 6, 3, 1, 3, 3, 6, 3, 3, 5, 10, 2, 3, 11, 9, 1,
 2, 3, 4, 1, 5, 4, 1, 2, 7, 2, 5, 5, 1, 2, 11, 5, 1, 3, 4, 6, 1, 4, 1, 10, 3, 2,
 2, 1, 4, 3, 4, 3, 2, 3, 5, 1, 7, 3, 1, 4, 1, 4, 1, 3, 3, 6, 4, 1, 3, 6, 2, 3,
 7, 4, 1, 8, 7, 5, 4, 4, 3, 5, 1, 3, 5, 7, 3, 5, 3, 5, 1, 3, 5, 1, 4, 1, 7, 1,
 3, 3, 6, 1, 3, 6, 3, 2, 1, 3, 5, 6, 1, 3, 4, 2, 4, 3, 4, 5, 1, 3, 5, 1, 4, 3,
 6, 6, 3, 1, 3, 4, 4, 7, 7, 1, 3, 4, 7, 3, 8, 2, 2, 3, 4, 1, 3, 10, 2, 4, 1, 2,
 6, 8, 2, 7, 4, 1, 9, 6, 1, 1, 6, 2, 5, 7, 5, 1, 1, 2, 2, 5, 1, 6, 1, 2, 1, 2,
 4, 4, 8, 5, 10, 1, 3, 1, 3, 3, 4, 6, 5, 1,

 5, 5, 1, 2, 5, 6, 2, 1, 3, 8, 2, 7, 2, 1, 1, 4, 2, 5, 8, 6, 1, 3, 5, 2, 4, 3,
 1, 2, 2, 8, 2, 2, 1, 4, 4, 5, 1, 3, 2, 3, 7, 3, 3, 8, 5, 6, 2, 1, 3, 4, 4, 1,
 2, 3, 1, 5, 1, 1, 5, 3, 3, 4, 3, 8, 4, 1, 1, 3, 5, 3, 3, 6, 1, 3, 3, 4, 7, 2,
 8, 2, 3, 4, 4, 3, 2, 7, 5, 2, 4, 6, 8, 1, 5, 5, 2, 3, 4, 3, 4, 1, 4, 4, 2, 6,
 8, 1, 5, 2, 2, 1, 8, 3, 5, 2, 3, 4, 4, 4, 1, 4, 5, 5, 2, 3, 4, 2, 4, 6, 1, 13,
 7, 1, 3, 7, 1, 7, 3, 2, 4, 1, 3, 4, 7, 4, 4, 5, 2, 3, 1, 4, 1, 7, 4, 3, 7, 1,
 5, 8, 8, 1, 2, 2, 2, 3, 4, 7, 4, 2, 3, 5, 1, 3, 2, 4, 2, 3, 5, 7, 2, 3, 7, 3,
 6, 1, 2, 2, 3, 1, 4, 5, 3, 7, 6, 1, 3, 4, 1, 8, 4, 1, 6, 7, 2, 5, 1, 3, 1, 7,
 3, 8, 5, 7, 4, 1, 3, 4, 3, 7, 3, 4, 1, 3, 1, 2, 1, 6, 6, 1, 1, 4, 2, 9, 2, 3,
 6, 1, 1, 8, 1, 2, 2, 4, 2, 3, 1, 2, 4, 2, 4, 6, 4, 3, 4, 2, 6, 3, 5, 6, 1, 7,
 4, 5, 6, 3, 1, 1, 7, 2, 4, 1, 3, 4, 1, 3, 1, 1, 8, 7, 6, 7, 1, 4, 2, 2, 1, 8,
 7, 2, 3, 1, 3, 1, 2, 4, 1, 5, 4, 5, 2, 4, 2, 13, 3, 2, 5, 2, 1, 4, 1, 5, 2, 4,
 4, 2, 4, 4, 10, 5, 3, 5, 3, 5, 4, 3, 1, 3, 4, 2, 

 6, 1, 7, 2, 3, 7, 1, 1, 9, 1, 4, 2, 7, 2, 4, 4, 2, 8, 3, 1, 3, 4, 4, 3, 6, 1,
 3, 4, 10, 4, 4, 3, 5, 1, 3, 3, 6, 4, 8, 4, 3, 3, 8, 1, 4, 3, 5, 5, 4, 3, 2, 2,
 2, 1, 8, 1, 3, 1, 7, 6, 2, 1, 8, 3, 7, 10, 2, 3, 6, 2, 8, 1, 2, 3, 3, 8, 10, 2,
 5, 6, 3, 7, 8, 1, 1, 3, 2, 1, 5, 12, 9, 3, 4, 1, 1, 9, 5, 1, 3, 3, 2, 3, 10, 6,
 4, 3, 4, 1, 3, 4, 2, 4, 3, 11, 6, 3, 2, 1, 7, 1, 4, 3, 7, 6, 4, 4, 5, 2, 1, 1,
 4, 2, 7, 2, 3, 2, 2, 8, 1, 1, 3, 4, 3, 4, 6, 1, 1, 7, 2, 4, 1, 1, 6, 2, 4, 1,
 6, 4, 4, 3, 2, 13, 4, 2, 3, 1, 1, 3, 1, 3, 6, 2, 4, 4, 5, 2, 2, 4, 1, 7, 4, 3,
 5, 3, 1, 3, 4, 7, 7, 2, 8, 3, 4, 3, 1, 4, 3, 5, 7, 2, 3, 7, 6, 6, 4, 2, 3, 6,
 1, 1, 3, 2, 4, 1, 1, 1, 2, 4, 4, 6, 3, 8, 2, 1, 8, 4, 5, 5, 4, 5, 2, 1, 5, 4,
 2, 1, 2, 1, 7, 3, 3, 1, 1, 1, 1, 3, 3, 2, 2, 4, 7, 1, 1, 2, 6, 6, 1, 1, 4, 4,
 5, 4, 3, 1, 1, 1, 1, 3, 3, 3, 4, 4, 2, 1, 2, 1, 3, 4, 3, 3, 3, 6, 3, 3, 2, 1,
 1, 4, 1, 1, 1, 4, 2, 1, 5, 5, 2, 1, 5, 2, 1, 4, 3, 3, 9, 5, 1, 8, 2, 4, 1, 1,
 1, 2, 1, 1, 3, 6, 1, 6, 1, 2, 4, 2, 3, 3, 12

 5, 1, 3, 4, 4, 9, 5, 4, 1, 4, 2, 4, 4, 3, 5, 1, 1, 5, 3, 4, 7, 7, 5, 5, 1, 2,
 1, 3, 5, 2, 4, 1, 2, 8, 1, 1, 2, 3, 1, 1, 2, 7, 1, 3, 1, 1, 3, 2, 7, 1, 1, 5,
 1, 4, 2, 2, 1, 5, 2, 5, 1, 5, 1, 5, 5, 2, 3, 4, 4, 6, 5, 6, 2, 1, 4, 1, 4, 4,
 4, 3, 9, 1, 3, 4, 11, 7, 7, 2, 1, 5, 8, 7, 1, 6, 1, 2, 3, 9, 8, 1, 9, 7, 4, 8,
 2, 2, 1, 2, 3, 6, 1, 1, 5, 6, 2, 1, 5, 3, 1, 4, 3, 2, 4, 1, 1, 1, 6, 4, 2, 4,
 8, 1, 7, 7, 1, 1, 2, 1, 3, 2, 1, 3, 1, 1, 5, 6, 1, 4, 2, 1, 3, 1, 2, 3, 2, 4,
 1, 1, 1, 3, 4, 8, 2, 4, 2, 5, 1, 2, 3, 4, 1, 2, 1, 6, 1, 4, 1, 8, 3, 1, 6, 11,
 2, 1, 2, 1, 2, 1, 2, 1, 1, 4, 1, 1, 5, 2, 4, 2, 4, 6, 1, 6, 3, 11, 7, 4, 2, 2,
 8, 1, 4, 3, 4, 13, 7, 1, 1, 6, 2, 4, 1, 1, 1, 3, 4, 6, 2, 6, 3, 2, 2, 11, 1, 2,
 3, 5, 2, 3, 6, 3, 7, 6, 1, 3, 5, 6, 3, 2, 2, 8, 2, 1, 2, 2, 1, 2, 2, 1, 6, 4,
 5, 9, 2, 1, 2, 1, 1, 7, 1, 8, 2, 4, 1, 1, 1, 4, 2, 2, 1, 6, 2, 5, 10, 6, 1, 7,
 2, 4, 2, 3, 7, 1, 3, 5, 1, 1, 5, 4, 3, 2, 4, 1, 1, 1, 8, 1, 4, 2, 2, 3, 2, 5,
 2, 2, 2, 5, 1, 3, 8, 8, 3, 4, 2, 2, 8, 1, 4, 

 3, 5, 1, 5, 3, 5, 7, 7, 4, 4, 1, 2, 8, 3, 8, 1, 8, 3, 11, 3, 8, 1, 3, 7, 2, 7,
 6, 3, 3, 4, 4, 2, 3, 7, 7, 9, 2, 3, 7, 5, 1, 2, 3, 3, 8, 3, 5, 5, 7, 3, 7, 6,
 2, 3, 7, 5, 1, 8, 5, 8, 1, 3, 1, 2, 5, 9, 1, 2, 5, 2, 10, 5, 3, 4, 5, 8, 2, 4,
 10, 3, 10, 2, 4, 1, 5, 1, 3, 7, 4, 3, 4, 2, 3, 5, 1, 3, 3, 13, 2, 3, 7, 9, 2,
 5, 5, 1, 4, 5, 7, 2, 3, 5, 4, 2, 3, 6, 1, 9, 1, 3, 5, 2, 3, 11, 5, 2, 3, 8, 1,
 5, 3, 5, 7, 1, 4, 9, 2, 1, 8, 1, 3, 5, 5, 4, 7, 5, 6, 2, 13, 9, 1, 3, 3, 5, 1,
 4, 7, 5, 1, 3, 5, 1, 4, 2, 3, 4, 6, 2, 4, 3, 6, 6, 1, 8, 1, 2, 3, 7, 2, 3, 5,
 10, 10, 1, 3, 9, 4, 7, 5, 5, 4, 10, 2, 3, 10, 7, 2, 3, 5, 1, 4, 7, 8, 2, 6, 2,
 5, 6, 1, 3, 9, 5, 2, 3, 9, 7, 3, 4, 5, 1, 3, 2, 10, 6, 2, 3, 5, 1, 4, 1, 4, 7,
 3, 3, 3, 3, 10, 3, 3, 8, 1, 3, 2, 4, 5, 7, 3, 9, 12, 2, 5, 5, 12, 1, 2, 4, 6,
 1, 6, 4, 1, 5, 1, 2, 4, 2, 6, 10, 1, 2, 2, 3, 5, 1, 4, 5, 2, 4, 3, 6, 4, 2, 3,
 1, 3, 7, 3, 2, 2, 3, 9, 1, 3, 2, 1, 2, 3, 4, 2, 3, 4, 1, 3, 5, 7, 4, 7, 3, 2,
 13, 5, 12, 9, 4, 2, 1, 4, 2, 1, 5, 2, 6, 5, 

 5, 8, 2, 3, 4, 1, 3, 5, 1, 5, 3, 4, 8, 10, 2, 5, 5, 10, 10, 1, 3, 6, 10, 3, 12,
 2, 5, 11, 2, 3, 6, 1, 5, 6, 1, 4, 6, 3, 1, 8, 3, 3, 1, 4, 2, 3, 3, 9, 3, 1, 5,
 3, 7, 2, 5, 1, 3, 4, 4, 5, 3, 2, 4, 9, 1, 1, 11, 1, 1, 10, 1, 1, 4, 3, 1, 3, 1,
 2, 5, 9, 4, 3, 4, 4, 4, 3, 3, 6, 1, 3, 1, 5, 2, 3, 6, 5, 7, 1, 6, 3, 3, 4, 7,
 4, 4, 2, 2, 3, 9, 1, 3, 2, 5, 6, 3, 6, 2, 8, 5, 2, 6, 1, 9, 7, 4, 9, 6, 1, 3,
 3, 3, 11, 7, 6, 2, 1, 2, 3, 3, 2, 5, 4, 8, 1, 3, 5, 4, 3, 6, 6, 8, 2, 3, 10, 1,
 4, 4, 1, 4, 4, 3, 4, 3, 7, 7, 3, 5, 4, 1, 10, 4, 2, 7, 3, 10, 8, 1, 4, 2, 1, 7,
 2, 3, 4, 2, 3, 4, 2, 3, 8, 5, 1, 2, 4, 2, 1, 8, 2, 6, 7, 3, 8, 1, 10, 7, 3, 4,
 2, 1, 7, 6, 12, 1, 7, 4, 5, 4, 3, 7, 1, 4, 1, 5, 6, 2, 4, 2, 6, 6, 7, 5, 3, 1,
 3, 2, 3, 8, 4, 3, 1, 2, 6, 4, 1, 3, 3, 4, 4, 3, 5, 1, 4, 3, 4, 3, 13, 6, 3, 3,
 5, 7, 9, 4, 1, 2, 6, 4, 3, 8, 4, 5, 9, 4, 10, 1, 3, 5, 4, 1, 5, 4, 1, 6, 4, 2,
 5, 6, 1, 3, 6, 2, 2, 11, 5, 1, 4, 7, 1, 1, 4, 7, 1, 9, 3, 3, 3, 4, 5, 3, 4, 5,
 1, 3, 10, 5, 4, 3, 4, 2, 3, 5, 8, 4, 

 2, 2, 5, 2, 5, 7, 2, 11, 5, 1, 3, 2, 3, 2, 2, 4, 1, 2, 2, 2, 9, 7, 1, 6, 6, 8,
 2, 10, 11, 1, 2, 3, 1, 7, 1, 3, 5, 5, 1, 10, 2, 6, 1, 5, 3, 8, 2, 3, 10, 1, 3,
 5, 3, 5, 6, 6, 1, 3, 2, 2, 9, 4, 1, 3, 3, 7, 7, 2, 6, 4, 4, 4, 2, 7, 13, 14, 1,
 2, 4, 2, 4, 7, 3, 3, 3, 7, 1, 9, 2, 2, 6, 2, 3, 3, 11, 2, 10, 8, 1, 2, 2, 6, 3,
 8, 2, 4, 4, 6, 2, 4, 1, 1, 8, 5, 2, 3, 4, 9, 1, 1, 7, 7, 8, 1, 7, 3, 5, 5, 8,
 2, 2, 8, 1, 4, 3, 4, 4, 6, 10, 2, 6, 1, 1, 5, 3, 7, 2, 1, 6, 10, 11, 2, 3, 7,
 7, 1, 2, 2, 2, 4, 5, 2, 5, 2, 2, 4, 2, 3, 11, 1, 2, 6, 4, 2, 6, 2, 7, 3, 7, 2,
 1, 15, 2, 8, 1, 1, 2, 3, 9, 1, 3, 3, 7, 1, 3, 2, 2, 1, 9, 4, 1, 2, 8, 6, 1, 6,
 2, 3, 4, 1, 6, 2, 5, 2, 2, 1, 2, 5, 4, 4, 3, 6, 2, 3, 3, 5, 4, 1, 2, 4, 3, 1,
 1, 5, 5, 3, 11, 11, 1, 4, 1, 3, 4, 6, 2, 3, 5, 1, 4, 2, 1, 7, 5, 1, 4, 2, 1, 4,
 1, 6, 5, 2, 4, 3, 4, 2, 4, 9, 1, 3, 5, 9, 2, 3, 5, 4, 5, 2, 1, 1, 4, 2, 3, 1,
 9, 1, 4, 1, 8, 4, 4, 1, 6, 3, 4, 2, 3, 2, 4, 1, 3, 1, 4, 4, 7, 9, 3, 6, 7, 6,
 1, 1, 4, 3, 2, 2, 4, 6, 4, 5, 7, 5, 1, 1,

 2, 6, 5, 5, 2, 9, 6, 1, 7, 4, 3, 6, 2, 7, 5, 1, 3, 1, 4, 4, 2, 10, 8, 1, 3, 3,
 6, 4, 10, 4, 3, 4, 6, 1, 4, 2, 12, 1, 3, 10, 7, 1, 7, 1, 2, 3, 6, 2, 3, 6, 1,
 4, 4, 6, 5, 3, 2, 4, 1, 3, 2, 4, 1, 9, 1, 3, 6, 3, 7, 3, 4, 1, 4, 4, 2, 4, 1,
 3, 1, 3, 5, 2, 11, 6, 8, 4, 10, 1, 4, 2, 3, 5, 9, 2, 7, 3, 4, 1, 4, 3, 5, 5, 8,
 4, 3, 3, 2, 3, 11, 5, 1, 3, 5, 9, 3, 2, 8, 2, 5, 5, 10, 1, 5, 4, 1, 4, 4, 1, 4,
 5, 3, 9, 2, 9, 1, 3, 4, 2, 4, 4, 10, 8, 1, 4, 5, 4, 2, 3, 1, 7, 2, 3, 1, 3, 3,
 7, 2, 1, 7, 1, 2, 1, 4, 1, 8, 1, 4, 1, 3, 8, 1, 7, 2, 3, 11, 1, 10, 3, 8, 2, 3,
 5, 1, 5, 1, 4, 5, 2, 4, 1, 5, 2, 3, 2, 3, 4, 1, 6, 8, 11, 2, 4, 1, 5, 1, 7, 4,
 3, 9, 2, 3, 10, 5, 2, 8, 1, 1, 5, 7, 2, 6, 2, 3, 5, 6, 4, 3, 5, 1, 5, 1, 3, 5,
 10, 8, 2, 3, 10, 1, 1, 8, 2, 7, 2, 3, 4, 1, 4, 10, 2, 3, 3, 5, 7, 1, 3, 9, 3,
 4, 3, 1, 3, 7, 2, 1, 5, 4, 3, 3, 6, 6, 2, 3, 6, 3, 6, 3, 7, 1, 3, 3, 7, 7, 3,
 3, 2, 7, 6, 3, 11, 1, 9, 4, 4, 5, 3, 1, 6, 1, 5, 1, 7, 1, 2, 4, 4, 2, 10, 2, 5,
 10, 1, 2, 5, 4, 1, 3, 7, 1, 1, 9, 3, 10

 1, 5, 3, 5, 1, 1, 7, 4, 5, 3, 6, 5, 1, 8, 5, 3, 4, 1, 2, 2, 3, 5, 4, 8, 6, 1,
 5, 7, 3, 3, 7, 4, 4, 1, 2, 2, 4, 1, 4, 3, 6, 8, 2, 3, 3, 1, 4, 2, 5, 4, 6, 7,
 3, 5, 2, 3, 6, 8, 2, 3, 6, 1, 5, 3, 5, 1, 4, 2, 3, 8, 6, 2, 3, 3, 1, 4, 7, 7,
 3, 3, 5, 1, 3, 7, 4, 10, 5, 1, 3, 6, 5, 5, 4, 1, 10, 8, 3, 4, 7, 1, 3, 3, 4, 4,
 1, 2, 4, 6, 7, 4, 1, 1, 7, 14, 7, 3, 1, 3, 4, 2, 4, 1, 4, 2, 8, 1, 6, 4, 9, 8,
 2, 12, 5, 6, 3, 4, 11, 11, 1, 3, 4, 2, 2, 8, 11, 1, 8, 10, 1, 4, 2, 6, 1, 3, 5,
 8, 4, 3, 7, 2, 4, 3, 5, 4, 3, 6, 2, 3, 3, 4, 5, 3, 5, 5, 3, 4, 3, 6, 6, 2, 5,
 1, 3, 4, 4, 2, 4, 2, 9, 1, 2, 3, 2, 5, 8, 9, 1, 2, 1, 4, 5, 2, 4, 5, 1, 6, 1,
 2, 5, 7, 1, 3, 3, 2, 3, 7, 5, 3, 5, 2, 3, 4, 1, 4, 5, 6, 2, 7, 6, 6, 1, 3, 2,
 2, 4, 2, 2, 8, 4, 3, 5, 2, 5, 3, 6, 2, 12, 1, 7, 2, 3, 5, 4, 2, 4, 7, 4, 1, 6,
 2, 2, 5, 9, 2, 8, 11, 2, 1, 4, 14, 1, 2, 7, 2, 5, 1, 11, 9, 2, 7, 1, 4, 2, 8,
 2, 1, 3, 1, 1, 7, 1, 1, 6, 1, 1, 6, 3, 1, 1, 6, 1, 1, 5, 1, 2, 3, 5, 2, 5, 5,
 3, 4, 12, 2, 5, 3, 11, 7, 10, 1, 4, 4, 5, 5, 

 13, 2, 3, 9, 6, 6, 1, 3, 4, 6, 8, 2, 9, 2, 1, 3, 9, 11, 2, 3, 5, 1, 5, 5, 5, 2,
 2, 1, 1, 3, 7, 1, 3, 7, 2, 5, 4, 1, 9, 3, 8, 6, 1, 3, 7, 2, 3, 11, 1, 2, 7, 4,
 2, 7, 5, 7, 1, 4, 7, 4, 6, 3, 4, 3, 7, 1, 4, 5, 8, 2, 5, 5, 1, 5, 3, 5, 5, 5,
 2, 3, 4, 4, 5, 6, 1, 3, 3, 4, 2, 3, 4, 1, 1, 7, 6, 1, 6, 5, 3, 5, 1, 2, 5, 3,
 2, 7, 11, 6, 5, 3, 2, 3, 4, 1, 3, 5, 4, 4, 3, 3, 1, 2, 3, 9, 4, 2, 3, 7, 6, 4,
 3, 3, 2, 5, 6, 1, 3, 2, 6, 2, 2, 5, 4, 2, 3, 5, 2, 3, 5, 1, 3, 1, 4, 1, 4, 2,
 4, 4, 3, 6, 5, 2, 3, 6, 1, 1, 5, 3, 4, 2, 2, 5, 1, 7, 3, 5, 7, 6, 2, 3, 7, 6,
 1, 1, 3, 4, 5, 3, 5, 1, 5, 4, 2, 1, 5, 1, 5, 8, 2, 4, 8, 1, 3, 1, 1, 3, 3, 2,
 5, 1, 1, 4, 1, 2, 1, 2, 1, 2, 4, 5, 11, 1, 2, 10, 2, 2, 1, 5, 3, 2, 3, 7, 1, 8,
 3, 1, 4, 4, 1, 6, 7, 1, 4, 3, 2, 4, 1, 4, 3, 2, 5, 1, 4, 2, 2, 2, 4, 1, 1, 1,
 6, 1, 3, 1, 2, 4, 3, 5, 1, 2, 8, 4, 2, 6, 3, 4, 1, 7, 1, 6, 1, 1, 6, 3, 1, 2,
 6, 1, 1, 4, 8, 2, 3, 5, 4, 7, 1, 3, 2, 5, 1, 5, 2, 4, 4, 6, 1, 3, 2, 11, 1, 5,
 5, 4, 4, 5, 1, 3, 2, 1, 5, 8, 1, 4, 2, 2, 1,

 3, 5, 6, 1, 3, 1, 5, 2, 3, 5, 6, 2, 5, 1, 8, 1, 3, 10, 3, 4, 4, 9, 4, 3, 6, 6,
 13, 1, 4, 2, 4, 13, 7, 5, 5, 5, 8, 9, 1, 2, 2, 6, 2, 5, 3, 1, 5, 11, 2, 4, 2,
 1, 7, 1, 2, 9, 1, 6, 4, 1, 7, 4, 2, 3, 4, 1, 3, 10, 6, 2, 4, 8, 6, 1, 3, 12, 2,
 3, 5, 2, 7, 4, 4, 1, 3, 5, 1, 4, 5, 13, 1, 6, 3, 8, 1, 4, 10, 3, 6, 7, 3, 5, 5,
 1, 4, 7, 1, 5, 2, 3, 7, 15, 7, 10, 4, 4, 1, 2, 3, 6, 2, 2, 7, 1, 4, 7, 7, 8, 2,
 4, 11, 1, 2, 4, 2, 4, 1, 4, 1, 2, 2, 4, 2, 4, 1, 8, 1, 1, 5, 5, 2, 8, 9, 3, 9,
 1, 5, 5, 1, 5, 5, 5, 8, 2, 3, 10, 1, 3, 10, 7, 4, 1, 2, 4, 5, 2, 3, 3, 2, 6,
 11, 1, 1, 3, 1, 4, 3, 5, 2, 8, 4, 4, 4, 2, 5, 4, 4, 4, 1, 3, 2, 2, 4, 3, 5, 5,
 7, 3, 4, 6, 2, 7, 3, 3, 10, 9, 2, 5, 8, 1, 7, 1, 3, 6, 2, 3, 7, 8, 1, 9, 2, 3,
 1, 4, 5, 3, 2, 8, 10, 4, 6, 2, 4, 4, 1, 4, 7, 1, 7, 1, 3, 3, 4, 4, 1, 4, 7, 2,
 7, 3, 8, 6, 2, 4, 1, 4, 1, 2, 2, 4, 1, 2, 3, 6, 2, 3, 5, 5, 7, 2, 5, 1, 4, 2,
 3, 10, 5, 1, 2, 4, 3, 1, 6, 1, 3, 10, 9, 7, 4, 3, 3, 5, 3, 8, 4, 9, 8, 1, 3, 2,
 3, 5, 3, 1, 10, 7, 2, 3, 5, 5, 1, 2, 6,

 2, 2, 5, 7, 4, 2, 9, 8, 1, 4, 4, 9, 4, 8, 1, 2, 3, 4, 8, 2, 5, 6, 1, 2, 4, 3,
 3, 6, 1, 6, 5, 2, 3, 8, 4, 3, 2, 7, 7, 2, 3, 5, 1, 8, 4, 2, 5, 2, 1, 4, 9, 2,
 3, 5, 4, 1, 2, 3, 8, 2, 5, 2, 3, 3, 5, 4, 2, 5, 6, 4, 4, 6, 1, 4, 2, 2, 3, 3,
 9, 1, 10, 8, 1, 5, 9, 3, 11, 11, 2, 4, 5, 1, 3, 1, 8, 1, 4, 10, 11, 1, 2, 5, 2,
 4, 1, 4, 5, 4, 3, 12, 1, 4, 3, 13, 13, 2, 3, 8, 6, 8, 2, 3, 6, 2, 9, 9, 3, 4,
 1, 3, 5, 1, 2, 8, 8, 3, 4, 8, 8, 2, 3, 5, 1, 3, 5, 4, 9, 2, 4, 2, 9, 1, 8, 1,
 3, 4, 8, 3, 4, 2, 2, 4, 1, 3, 1, 4, 2, 3, 3, 3, 9, 1, 3, 5, 3, 4, 4, 11, 6, 1,
 2, 3, 3, 5, 5, 1, 3, 5, 7, 11, 2, 6, 2, 4, 8, 1, 4, 6, 2, 3, 8, 5, 1, 9, 7, 4,
 3, 4, 7, 2, 3, 5, 9, 1, 4, 11, 2, 4, 7, 3, 2, 3, 6, 2, 3, 4, 4, 6, 4, 4, 4, 1,
 3, 6, 4, 3, 3, 4, 6, 4, 1, 7, 7, 1, 1, 7, 4, 2, 6, 7, 12, 3, 4, 6, 1, 3, 3, 5,
 7, 5, 1, 3, 1, 2, 8, 2, 3, 5, 1, 6, 9, 2, 5, 2, 4, 5, 1, 3, 3, 4, 10, 2, 3, 6,
 1, 3, 6, 4, 3, 6, 1, 7, 4, 2, 3, 9, 1, 6, 4, 5, 6, 2, 9, 10, 3, 4, 5, 1, 2, 5,
 4, 5, 7, 5, 2, 3, 12, 1, 3, 9, 5, 5, 4, 5, 

 4, 3, 6, 2, 4, 4, 4, 4, 5, 2, 4, 5, 2, 5, 8, 1, 3, 3, 3, 6, 2, 4, 3, 6, 2, 4,
 2, 5, 6, 1, 5, 2, 5, 1, 4, 4, 4, 2, 7, 1, 4, 6, 1, 7, 4, 3, 7, 1, 6, 2, 7, 1,
 3, 5, 6, 1, 6, 2, 4, 10, 8, 2, 3, 1, 6, 1, 6, 12, 4, 3, 5, 2, 6, 4, 2, 5, 1, 2,
 4, 3, 6, 1, 10, 5, 3, 5, 1, 5, 5, 1, 4, 4, 4, 11, 10, 1, 4, 2, 4, 4, 6, 3, 6,
 1, 2, 3, 5, 1, 4, 3, 5, 2, 4, 10, 9, 1, 6, 1, 2, 5, 5, 1, 3, 9, 5, 2, 3, 3, 5,
 2, 8, 2, 1, 8, 1, 1, 8, 1, 3, 6, 3, 4, 2, 7, 2, 8, 8, 2, 3, 3, 11, 1, 8, 4, 1,
 3, 3, 9, 4, 2, 5, 2, 5, 5, 2, 2, 4, 2, 4, 1, 9, 8, 4, 3, 5, 2, 5, 1, 4, 2, 4,
 1, 2, 3, 4, 2, 8, 2, 6, 2, 8, 2, 3, 5, 7, 3, 6, 5, 4, 3, 5, 1, 4, 1, 1, 4, 1,
 5, 1, 4, 13, 7, 5, 4, 5, 4, 1, 3, 3, 2, 1, 4, 2, 10, 7, 6, 4, 7, 3, 5, 4, 4, 1,
 9, 6, 3, 4, 5, 10, 2, 6, 3, 1, 3, 5, 3, 4, 2, 13, 5, 1, 4, 4, 5, 8, 7, 1, 7, 1,
 5, 4, 4, 2, 9, 3, 4, 1, 3, 2, 8, 3, 6, 8, 2, 5, 7, 5, 7, 2, 5, 1, 2, 3, 3, 1,
 2, 5, 3, 4, 7, 2, 3, 3, 1, 2, 4, 4, 3, 2, 2, 6, 7, 8, 2, 1, 7, 1, 5, 3, 1, 12,
 5, 2, 5, 5, 6, 2, 8, 3, 9, 3, 5, 10, 6, 1,

 3, 4, 6, 2, 1, 2, 3, 8, 2, 4, 5, 4, 2, 3, 7, 1, 3, 5, 3, 5, 3, 6, 2, 2, 6, 5,
 10, 7, 3, 4, 4, 7, 1, 1, 5, 7, 4, 7, 4, 8, 1, 3, 3, 2, 7, 5, 2, 2, 1, 2, 3, 5,
 5, 3, 6, 2, 5, 2, 3, 5, 5, 1, 2, 4, 4, 3, 8, 1, 2, 3, 4, 5, 1, 3, 4, 2, 5, 3,
 9, 5, 4, 2, 5, 5, 1, 3, 2, 6, 3, 4, 6, 1, 3, 4, 9, 3, 3, 6, 6, 2, 3, 4, 3, 5,
 1, 3, 7, 4, 5, 4, 3, 4, 1, 5, 2, 4, 4, 1, 3, 1, 2, 3, 3, 8, 1, 3, 2, 3, 1, 4,
 5, 4, 4, 8, 5, 3, 5, 8, 2, 1, 7, 5, 5, 4, 1, 2, 1, 4, 1, 4, 11, 2, 3, 4, 1, 3,
 7, 5, 1, 3, 5, 6, 3, 5, 4, 7, 3, 5, 1, 2, 1, 5, 5, 3, 4, 1, 4, 1, 5, 1, 2, 2,
 4, 5, 8, 4, 1, 2, 3, 5, 5, 1, 4, 6, 1, 2, 7, 4, 2, 5, 1, 2, 4, 4, 1, 3, 2, 5,
 4, 5, 4, 1, 2, 4, 6, 1, 10, 7, 2, 11, 2, 3, 8, 1, 3, 2, 2, 7, 9, 4, 8, 10, 2,
 3, 9, 1, 4, 4, 5, 8, 2, 4, 10, 4, 3, 5, 1, 3, 6, 2, 2, 4, 5, 9, 3, 7, 11, 7, 1,
 3, 1, 4, 2, 7, 4, 4, 3, 4, 3, 1, 10, 12, 1, 1, 10, 3, 4, 11, 3, 1, 2, 5, 3, 4,
 3, 4, 2, 3, 1, 4, 3, 4, 2, 3, 4, 10, 5, 2, 4, 4, 12, 2, 5, 5, 2, 3, 7, 3, 1, 3,
 4, 5, 7, 1, 4, 3, 7, 1, 3, 4, 11, 2, 3, 4, 1

 1, 5, 3, 4, 1, 1, 4, 4, 4, 4, 14, 6, 1, 4, 3, 4, 3, 10, 2, 3, 3, 1, 9, 3, 5, 6,
 5, 2, 3, 4, 6, 2, 5, 3, 1, 3, 6, 3, 11, 7, 6, 4, 4, 3, 4, 1, 4, 6, 2, 9, 11, 5,
 2, 3, 8, 1, 6, 4, 3, 6, 8, 4, 3, 8, 2, 1, 1, 5, 3, 4, 3, 5, 3, 4, 2, 5, 8, 1,
 1, 4, 1, 8, 1, 5, 2, 2, 6, 9, 2, 2, 3, 10, 6, 3, 2, 3, 3, 1, 3, 6, 5, 5, 5, 1,
 3, 4, 9, 5, 4, 5, 3, 7, 3, 4, 5, 3, 5, 3, 4, 4, 8, 1, 8, 4, 3, 7, 5, 2, 3, 4,
 5, 2, 3, 4, 12, 10, 5, 4, 3, 7, 1, 8, 7, 3, 4, 6, 3, 8, 9, 7, 2, 3, 7, 8, 2, 3,
 4, 2, 1, 5, 1, 3, 8, 6, 3, 4, 4, 3, 2, 3, 3, 1, 3, 4, 3, 7, 5, 1, 1, 4, 3, 4,
 5, 12, 5, 1, 3, 9, 10, 8, 4, 3, 3, 2, 3, 10, 1, 1, 5, 4, 2, 4, 6, 2, 3, 4, 1,
 3, 6, 4, 3, 4, 5, 3, 5, 2, 3, 5, 8, 1, 2, 2, 4, 3, 8, 5, 4, 3, 5, 4, 1, 3, 3,
 2, 3, 4, 8, 4, 3, 4, 2, 1, 3, 1, 1, 4, 1, 3, 8, 4, 4, 1, 4, 2, 4, 1, 5, 6, 7,
 2, 3, 4, 1, 4, 2, 4, 4, 5, 3, 4, 2, 3, 3, 3, 6, 1, 3, 6, 4, 1, 4, 3, 5, 3, 5,
 1, 3, 2, 3, 4, 4, 6, 4, 1, 4, 3, 4, 2, 1, 5, 3, 4, 8, 1, 2, 4, 4, 10, 5, 1, 5,
 2, 3, 3, 2, 2, 5, 7, 4, 1, 4, 1, 1, 1, 3, 1

 3, 4, 5, 5, 3, 10, 7, 11, 1, 10, 3, 8, 5, 2, 3, 4, 2, 3, 9, 2, 5, 9, 1, 10, 1,
 11, 9, 1, 5, 4, 2, 5, 6, 4, 5, 8, 4, 2, 3, 7, 14, 2, 5, 1, 1, 6, 11, 1, 4, 3,
 6, 2, 3, 5, 3, 3, 1, 8, 12, 1, 3, 3, 3, 3, 11, 1, 9, 6, 2, 4, 1, 4, 4, 5, 2, 4,
 8, 1, 3, 5, 2, 3, 3, 3, 3, 4, 3, 4, 4, 7, 3, 2, 4, 3, 1, 4, 6, 1, 9, 2, 7, 1,
 3, 6, 6, 9, 4, 3, 9, 2, 3, 1, 3, 6, 4, 1, 3, 3, 11, 7, 1, 8, 2, 7, 6, 1, 7, 2,
 3, 6, 1, 5, 1, 4, 1, 5, 6, 1, 3, 5, 4, 4, 1, 3, 5, 6, 3, 6, 5, 6, 4, 7, 8, 1,
 7, 1, 4, 3, 3, 6, 1, 3, 3, 4, 2, 6, 1, 2, 3, 5, 1, 2, 6, 2, 5, 3, 3, 5, 2, 3,
 3, 2, 2, 5, 1, 3, 3, 6, 6, 1, 2, 8, 1, 6, 8, 1, 9, 4, 2, 1, 6, 1, 2, 3, 5, 7,
 7, 5, 6, 6, 4, 3, 6, 3, 9, 5, 1, 4, 3, 3, 5, 6, 2, 4, 7, 2, 6, 1, 3, 1, 2, 5,
 4, 1, 9, 9, 1, 4, 4, 6, 1, 7, 7, 1, 3, 1, 8, 1, 2, 4, 1, 6, 5, 1, 7, 3, 3, 4,
 3, 2, 7, 2, 3, 5, 7, 2, 3, 3, 1, 3, 2, 2, 3, 4, 1, 2, 6, 9, 1, 6, 4, 6, 3, 10,
 3, 8, 4, 3, 3, 1, 8, 5, 8, 4, 3, 1, 3, 2, 5, 9, 6, 1, 2, 3, 3, 4, 2, 5, 2, 3,
 4, 1, 4, 4, 1, 3, 5, 1, 4, 1, 3, 5, 1, 4, 3, 4

 4, 1, 7, 8, 1, 2, 4, 2, 2, 1, 4, 4, 8, 2, 4, 6, 1, 4, 4, 4, 3, 6, 5, 10, 2, 4,
 2, 5, 10, 5, 3, 6, 1, 1, 4, 1, 4, 1, 12, 2, 3, 6, 5, 4, 6, 6, 4, 3, 5, 2, 8, 8,
 2, 3, 4, 1, 3, 5, 2, 4, 7, 6, 1, 3, 7, 2, 3, 2, 7, 6, 1, 1, 3, 7, 5, 1, 3, 4,
 5, 8, 1, 5, 4, 3, 4, 1, 4, 1, 5, 4, 2, 8, 2, 7, 3, 6, 10, 8, 1, 3, 2, 5, 1, 1,
 4, 1, 1, 5, 8, 3, 5, 2, 5, 1, 4, 8, 2, 2, 4, 1, 2, 1, 8, 6, 2, 2, 7, 7, 1, 3,
 5, 1, 1, 3, 1, 7, 4, 3, 5, 1, 4, 5, 6, 2, 3, 6, 3, 1, 2, 5, 4, 9, 1, 3, 5, 4,
 1, 5, 3, 9, 1, 4, 2, 3, 6, 5, 4, 2, 1, 3, 5, 3, 11, 9, 4, 12, 3, 1, 3, 5, 3, 4,
 8, 2, 3, 4, 1, 1, 4, 3, 5, 5, 2, 2, 5, 8, 3, 6, 5, 1, 2, 5, 10, 2, 3, 1, 9, 1,
 3, 4, 1, 2, 1, 5, 6, 2, 9, 1, 3, 8, 2, 4, 7, 4, 1, 3, 4, 4, 4, 1, 4, 8, 4, 3,
 5, 2, 4, 4, 6, 3, 1, 3, 3, 9, 2, 3, 9, 2, 3, 4, 1, 3, 7, 4, 8, 3, 5, 1, 9, 2,
 5, 5, 1, 2, 3, 8, 4, 3, 4, 4, 5, 2, 3, 4, 4, 6, 1, 4, 8, 1, 5, 1, 4, 3, 7, 3,
 1, 9, 1, 3, 3, 7, 1, 6, 4, 7, 6, 3, 9, 4, 4, 4, 4, 7, 3, 2, 5, 9, 1, 3, 9, 5,
 4, 6, 2, 4, 4, 1, 3, 3, 4, 3, 5, 4, 10, 2, 3

 3, 7, 3, 3, 5, 6, 3, 9, 1, 3, 7, 7, 4, 3, 5, 2, 3, 7, 1, 7, 5, 5, 2, 5, 4, 2,
 1, 3, 6, 6, 3, 3, 7, 2, 5, 3, 5, 5, 7, 9, 2, 3, 12, 1, 4, 7, 3, 5, 1, 6, 1, 3,
 7, 9, 9, 4, 3, 5, 1, 6, 2, 1, 3, 6, 3, 3, 4, 4, 2, 3, 2, 6, 2, 7, 2, 4, 4, 6,
 2, 4, 1, 3, 4, 3, 3, 8, 7, 1, 1, 3, 5, 1, 4, 1, 5, 5, 3, 4, 2, 3, 3, 8, 1, 7,
 3, 3, 1, 10, 5, 3, 10, 7, 3, 4, 8, 1, 2, 5, 4, 5, 2, 3, 6, 3, 10, 5, 6, 1, 5,
 8, 7, 3, 8, 5, 1, 3, 3, 5, 2, 8, 6, 2, 5, 1, 3, 3, 3, 5, 8, 2, 5, 4, 7, 4, 2,
 3, 5, 1, 5, 8, 3, 8, 2, 9, 7, 3, 6, 5, 8, 9, 7, 3, 4, 1, 1, 2, 7, 3, 7, 4, 7,
 1, 3, 8, 2, 3, 4, 1, 8, 3, 9, 2, 4, 1, 3, 4, 1, 4, 4, 7, 1, 5, 3, 4, 7, 4, 2,
 4, 1, 3, 1, 6, 2, 8, 2, 7, 4, 6, 5, 2, 1, 4, 5, 7, 3, 3, 4, 4, 3, 9, 2, 1, 3,
 7, 8, 2, 3, 7, 1, 4, 1, 5, 5, 3, 3, 5, 4, 4, 3, 8, 1, 3, 2, 3, 6, 7, 1, 8, 9,
 1, 3, 4, 1, 9, 3, 5, 1, 2, 3, 9, 7, 3, 9, 1, 3, 8, 4, 3, 5, 2, 4, 3, 10, 1, 6,
 4, 4, 5, 2, 4, 2, 1, 3, 5, 9, 2, 3, 3, 1, 3, 1, 9, 4, 1, 6, 7, 2, 4, 5, 2, 8,
 11, 1, 3, 11, 4, 2, 9, 2, 10, 1, 10, 2, 4, 1, 3, 

 10, 12, 1, 2, 5, 3, 4, 1, 2, 3, 6, 1, 1, 10, 2, 7, 5, 3, 4, 6, 2, 6, 9, 4, 1,
 5, 4, 2, 7, 1, 1, 1, 1, 1, 4, 1, 8, 1, 1, 7, 6, 2, 11, 2, 3, 4, 5, 2, 3, 5, 1,
 5, 8, 2, 9, 1, 6, 2, 2, 3, 2, 5, 6, 3, 6, 1, 4, 5, 5, 2, 4, 1, 3, 11, 5, 5, 3,
 1, 5, 1, 4, 5, 5, 7, 4, 5, 6, 5, 1, 2, 7, 5, 1, 3, 10, 7, 2, 4, 2, 12, 3, 5, 4,
 13, 6, 1, 4, 5, 1, 5, 2, 2, 8, 6, 1, 7, 2, 3, 7, 7, 1, 1, 4, 7, 2, 3, 7, 4, 1,
 8, 1, 1, 6, 5, 4, 4, 4, 3, 5, 4, 1, 3, 3, 7, 2, 9, 3, 10, 5, 9, 2, 2, 4, 4, 2,
 2, 7, 10, 2, 2, 1, 1, 7, 2, 1, 8, 2, 10, 5, 1, 5, 2, 5, 4, 3, 6, 2, 3, 5, 2, 6,
 1, 6, 1, 4, 1, 3, 7, 2, 4, 2, 1, 5, 3, 3, 7, 3, 4, 9, 7, 5, 3, 3, 7, 2, 3, 7,
 1, 2, 7, 3, 3, 5, 5, 2, 1, 3, 2, 3, 4, 2, 3, 2, 3, 5, 7, 1, 9, 2, 3, 12, 5, 1,
 2, 1, 4, 5, 5, 5, 1, 5, 4, 3, 3, 2, 4, 5, 1, 3, 1, 6, 4, 4, 4, 4, 1, 2, 1, 2,
 7, 7, 1, 3, 3, 2, 3, 4, 2, 5, 7, 3, 2, 4, 3, 1, 3, 3, 3, 4, 13, 12, 2, 7, 7, 3,
 3, 7, 2, 3, 5, 3, 6, 1, 4, 4, 3, 4, 11, 2, 7, 1, 4, 2, 4, 7, 4, 1, 9, 1, 1, 8,
 6, 2, 2, 3, 8, 6, 7, 2, 4, 1, 5, 3, 6, 

 3, 5, 4, 3, 7, 6, 1, 4, 2, 5, 3, 3, 1, 3, 11, 2, 3, 3, 7, 2, 3, 4, 5, 4, 1, 3,
 4, 4, 1, 4, 3, 3, 4, 4, 4, 1, 4, 5, 5, 6, 1, 7, 2, 5, 1, 6, 6, 7, 3, 7, 2, 4,
 1, 3, 6, 5, 4, 2, 3, 3, 5, 1, 8, 7, 3, 6, 2, 4, 1, 1, 6, 3, 9, 3, 4, 1, 2, 4,
 2, 3, 11, 8, 3, 5, 4, 8, 5, 6, 5, 2, 4, 7, 6, 1, 2, 5, 4, 5, 2, 1, 5, 1, 3, 5,
 1, 2, 4, 9, 2, 3, 7, 5, 1, 3, 6, 1, 8, 3, 5, 2, 4, 1, 3, 3, 11, 4, 7, 1, 4, 1,
 9, 6, 13, 6, 4, 3, 3, 1, 7, 10, 4, 3, 8, 4, 1, 4, 3, 4, 9, 3, 1, 5, 2, 3, 6, 6,
 1, 6, 6, 4, 6, 4, 1, 9, 4, 7, 1, 4, 3, 9, 5, 9, 8, 2, 4, 6, 3, 2, 1, 2, 4, 8,
 4, 2, 3, 7, 6, 1, 7, 7, 3, 4, 2, 6, 4, 3, 4, 1, 4, 1, 4, 1, 2, 3, 8, 1, 1, 5,
 5, 4, 6, 4, 1, 3, 6, 6, 3, 6, 1, 3, 13, 6, 4, 3, 5, 1, 2, 4, 7, 6, 3, 4, 4, 1,
 4, 1, 5, 1, 3, 4, 4, 3, 5, 1, 4, 10, 3, 1, 6, 1, 2, 6, 2, 8, 1, 3, 4, 1, 4, 4,
 4, 5, 1, 3, 8, 7, 1, 2, 6, 3, 1, 5, 1, 3, 3, 2, 4, 4, 4, 1, 7, 6, 1, 5, 3, 7,
 4, 4, 1, 3, 2, 2, 1, 9, 4, 4, 1, 4, 1, 8, 1, 4, 3, 4, 8, 5, 3, 4, 3, 3, 1, 3,
 5, 6, 3, 3, 5, 1, 5, 5, 1, 5, 3, 8, 1, 5, 5,

 9, 3, 8, 6, 6, 2, 4, 4, 2, 5, 3, 1, 4, 2, 6, 3, 5, 8, 9, 3, 6, 1, 1, 4, 2, 2,
 4, 2, 2, 5, 5, 2, 3, 7, 1, 3, 1, 6, 5, 7, 3, 5, 1, 3, 5, 5, 7, 3, 4, 2, 2, 3,
 1, 5, 3, 3, 4, 3, 5, 5, 2, 1, 4, 7, 2, 2, 3, 5, 1, 5, 1, 6, 8, 3, 4, 2, 3, 6,
 5, 1, 6, 4, 5, 4, 4, 3, 6, 4, 1, 2, 3, 4, 2, 7, 1, 3, 4, 11, 8, 3, 2, 3, 3, 2,
 3, 5, 1, 1, 8, 6, 1, 3, 4, 7, 2, 3, 7, 5, 1, 3, 3, 4, 3, 6, 4, 3, 5, 2, 3, 7,
 4, 6, 4, 1, 4, 1, 3, 9, 4, 6, 3, 2, 5, 3, 2, 5, 1, 5, 7, 3, 7, 3, 8, 6, 2, 5,
 4, 1, 3, 8, 3, 7, 4, 4, 7, 4, 2, 3, 3, 1, 4, 4, 3, 10, 4, 5, 4, 1, 2, 4, 4, 3,
 6, 2, 4, 5, 4, 3, 3, 1, 3, 3, 3, 5, 1, 3, 5, 3, 2, 5, 5, 3, 11, 4, 4, 3, 8, 2,
 3, 5, 1, 3, 8, 4, 5, 4, 4, 3, 5, 2, 3, 7, 8, 1, 2, 2, 5, 4, 1, 7, 2, 4, 3, 7,
 5, 1, 2, 3, 4, 8, 2, 2, 2, 5, 12, 2, 5, 3, 5, 1, 2, 1, 3, 5, 5, 7, 3, 7, 1, 4,
 5, 3, 5, 4, 2, 4, 3, 5, 1, 1, 4, 2, 2, 4, 2, 2, 2, 3, 5, 2, 3, 7, 1, 2, 3, 6,
 5, 7, 2, 3, 5, 1, 1, 4, 2, 3, 5, 2, 2, 3, 3, 4, 6, 5, 3, 4, 1, 1, 5, 1, 4, 8,
 3, 4, 3, 3, 1, 5, 6, 7, 3, 4, 5, 5, 4, 3, 4, 1,

 7, 5, 4, 4, 3, 5, 3, 2, 1, 5, 3, 1, 3, 1, 2, 1, 5, 3, 4, 1, 2, 4, 7, 6, 1, 3,
 5, 3, 5, 2, 1, 2, 4, 2, 4, 6, 1, 7, 3, 5, 8, 4, 3, 4, 1, 3, 7, 2, 5, 4, 8, 1,
 1, 2, 1, 4, 1, 3, 6, 2, 2, 2, 4, 3, 4, 1, 3, 1, 2, 3, 1, 4, 1, 4, 3, 3, 4, 1,
 3, 3, 1, 1, 3, 4, 3, 6, 3, 5, 1, 3, 6, 2, 4, 1, 4, 1, 3, 1, 2, 2, 1, 4, 1, 1,
 3, 3, 1, 1, 3, 2, 4, 6, 1, 5, 2, 3, 5, 1, 1, 3, 4, 2, 5, 1, 4, 1, 1, 3, 1, 3,
 3, 4, 8, 1, 3, 3, 5, 4, 5, 3, 8, 1, 3, 1, 7, 3, 6, 2, 3, 4, 6, 2, 8, 7, 3, 6,
 1, 1, 4, 1, 4, 1, 3, 1, 2, 1, 2, 1, 4, 1, 3, 3, 1, 1, 3, 5, 1, 6, 3, 2, 3, 7,
 3, 1, 5, 2, 3, 7, 1, 2, 1, 1, 1, 8, 8, 1, 3, 4, 2, 4, 1, 2, 4, 2, 1, 1, 2, 2,
 1, 3, 4, 2, 2, 3, 2, 6, 3, 1, 4, 4, 5, 1, 4, 1, 3, 4, 2, 6, 1, 2, 2, 4, 1, 4,
 4, 3, 2, 3, 4, 1, 3, 4, 1, 1, 5, 1, 1, 3, 2, 4, 5, 1, 4, 1, 1, 1, 6, 2, 6, 1,
 4, 1, 4, 1, 4, 4, 5, 1, 5, 3, 4, 1, 1, 4, 7, 1, 5, 1, 5, 3, 4, 1, 1, 2, 2, 3,
 3, 4, 2, 1, 4, 2, 5, 3, 1, 3, 3, 9, 2, 1, 3, 1, 7, 2, 4, 3, 3, 5, 1, 4, 4, 5,
 6, 2, 7, 1, 6, 9, 3, 8, 4, 1, 2, 2, 2, 3, 4, 4, 2, 

 3, 2, 1, 4, 4, 2, 8, 12, 1, 5, 1, 1, 1, 4, 7, 3, 4, 7, 7, 1, 3, 3, 1, 8, 3, 2,
 5, 8, 1, 3, 3, 1, 6, 3, 13, 2, 5, 2, 8, 1, 2, 4, 3, 3, 2, 4, 5, 4, 6, 3, 5, 1,
 1, 4, 1, 2, 12, 2, 3, 6, 2, 1, 3, 4, 3, 7, 2, 3, 7, 4, 4, 7, 2, 6, 1, 2, 1, 8,
 10, 2, 5, 1, 4, 5, 3, 3, 9, 5, 2, 4, 1, 3, 2, 4, 3, 3, 1, 10, 2, 3, 4, 2, 1, 1,
 8, 5, 8, 3, 6, 4, 2, 2, 5, 1, 6, 2, 4, 5, 3, 1, 2, 1, 3, 2, 10, 9, 1, 6, 4, 4,
 1, 4, 3, 8, 9, 3, 8, 5, 1, 12, 5, 5, 3, 5, 4, 8, 2, 3, 3, 1, 3, 3, 5, 5, 7, 4,
 7, 2, 1, 11, 3, 5, 1, 1, 9, 6, 1, 4, 9, 7, 2, 3, 4, 2, 3, 3, 2, 6, 1, 5, 1, 3,
 8, 10, 1, 6, 1, 3, 3, 5, 10, 1, 7, 1, 5, 4, 3, 5, 1, 7, 4, 2, 2, 5, 5, 4, 4, 1,
 6, 1, 7, 5, 6, 10, 6, 5, 4, 1, 5, 3, 4, 1, 5, 1, 1, 4, 3, 2, 10, 1, 5, 1, 1, 4,
 7, 2, 10, 1, 5, 1, 1, 3, 5, 2, 4, 9, 8, 2, 3, 7, 5, 8, 2, 3, 5, 1, 3, 3, 3, 4,
 8, 2, 3, 3, 1, 5, 1, 1, 9, 1, 4, 1, 5, 2, 4, 3, 5, 5, 1, 1, 4, 3, 4, 4, 1, 4,
 6, 2, 3, 4, 1, 2, 3, 5, 3, 5, 7, 1, 9, 5, 2, 1, 5, 1, 3, 3, 1, 8, 2, 6, 1, 2,
 6, 1, 5, 3, 5, 6, 4, 3, 4, 1, 4, 3, 1,

 4, 8, 1, 4, 11, 4, 3, 8, 5, 3, 10, 2, 2, 4, 1, 3, 4, 5, 1, 2, 5, 2, 6, 3, 10,
 10, 3, 9, 8, 9, 3, 3, 12, 3, 9, 2, 3, 6, 1, 4, 6, 1, 1, 4, 6, 4, 1, 5, 5, 4, 4,
 8, 4, 7, 3, 5, 3, 3, 4, 2, 5, 2, 5, 12, 4, 1, 4, 3, 9, 7, 3, 7, 2, 4, 1, 3, 4,
 3, 3, 3, 4, 7, 8, 2, 3, 7, 2, 1, 10, 2, 3, 8, 4, 4, 4, 6, 3, 5, 1, 3, 6, 3, 7,
 3, 4, 6, 7, 2, 3, 6, 4, 5, 5, 5, 2, 4, 4, 3, 7, 2, 3, 5, 9, 7, 5, 2, 4, 4, 2,
 1, 7, 1, 3, 5, 2, 5, 5, 6, 3, 8, 1, 3, 5, 6, 4, 5, 4, 3, 11, 4, 1, 1, 5, 10,
 10, 4, 2, 3, 4, 9, 2, 4, 4, 2, 1, 4, 5, 1, 3, 9, 15, 11, 5, 4, 6, 5, 1, 3, 1,
 12, 1, 2, 3, 4, 2, 8, 4, 5, 4, 4, 4, 1, 8, 6, 2, 3, 3, 1, 2, 2, 6, 3, 4, 2, 4,
 5, 2, 4, 1, 11, 1, 7, 3, 6, 1, 4, 1, 4, 5, 1, 3, 3, 4, 6, 3, 4, 6, 4, 3, 5, 1,
 8, 2, 3, 3, 2, 4, 6, 1, 10, 9, 5, 2, 2, 3, 8, 4, 5, 7, 10, 1, 9, 4, 3, 4, 2, 3,
 2, 3, 6, 1, 8, 3, 6, 11, 10, 4, 3, 4, 1, 6, 2, 4, 5, 1, 6, 1, 9, 4, 11, 2, 4,
 6, 2, 4, 5, 5, 1, 4, 9, 2, 6, 1, 2, 2, 4, 1, 3, 4, 13, 3, 10, 2, 4, 1, 4, 5, 8,
 1, 1, 4, 9, 1, 1, 5, 1, 3, 2, 5, 5, 4, 2, 

 3, 6, 6, 2, 2, 1, 4, 3, 13, 12, 2, 8, 9, 2, 4, 6, 4, 5, 2, 5, 2, 4, 5, 1, 3, 8,
 3, 9, 2, 5, 7, 1, 2, 6, 2, 4, 6, 1, 1, 5, 4, 3, 4, 10, 8, 4, 3, 8, 10, 2, 8, 1,
 2, 5, 5, 7, 3, 13, 8, 6, 2, 6, 4, 7, 4, 1, 3, 3, 3, 2, 3, 5, 2, 4, 2, 7, 2, 3,
 3, 1, 2, 3, 3, 2, 6, 8, 2, 5, 4, 1, 3, 5, 4, 4, 5, 3, 5, 3, 5, 1, 3, 5, 8, 4,
 5, 1, 3, 3, 3, 2, 8, 4, 8, 1, 3, 4, 3, 4, 6, 3, 3, 6, 6, 1, 5, 1, 2, 10, 6, 2,
 2, 1, 2, 2, 6, 1, 6, 6, 3, 3, 1, 4, 7, 2, 5, 4, 1, 4, 9, 10, 2, 2, 2, 8, 7, 3,
 5, 5, 3, 3, 5, 1, 5, 3, 3, 2, 3, 6, 3, 2, 8, 7, 1, 4, 2, 3, 6, 2, 8, 3, 6, 7,
 4, 4, 1, 8, 1, 2, 2, 2, 2, 5, 13, 4, 7, 1, 2, 2, 3, 7, 1, 4, 2, 4, 1, 5, 2, 3,
 5, 6, 3, 3, 5, 7, 2, 5, 2, 3, 1, 3, 4, 3, 7, 7, 2, 3, 5, 2, 7, 4, 1, 4, 1, 1,
 1, 3, 2, 3, 4, 4, 5, 2, 3, 4, 2, 8, 1, 1, 5, 9, 1, 3, 3, 4, 9, 2, 3, 7, 2, 10,
 1, 5, 4, 4, 3, 3, 6, 2, 5, 1, 2, 7, 2, 7, 1, 3, 2, 3, 3, 3, 4, 2, 3, 3, 2, 1,
 7, 1, 4, 4, 1, 9, 2, 1, 3, 2, 3, 4, 8, 6, 3, 5, 1, 7, 2, 4, 2, 4, 1, 4, 4, 3,
 5, 5, 1, 1, 4, 1, 4, 2, 3, 3, 6, 2, 1, 4, 

 3, 6, 2, 6, 1, 3, 4, 3, 4, 7, 2, 2, 3, 5, 7, 1, 2, 4, 2, 3, 4, 6, 4, 1, 2, 5,
 5, 6, 9, 1, 3, 5, 6, 7, 3, 4, 4, 1, 6, 4, 3, 6, 4, 3, 3, 9, 4, 1, 7, 4, 3, 6,
 10, 6, 5, 2, 1, 3, 2, 2, 7, 3, 1, 1, 2, 6, 1, 1, 4, 1, 3, 2, 5, 5, 1, 1, 3, 3,
 7, 4, 1, 9, 5, 3, 5, 1, 3, 4, 4, 4, 3, 1, 7, 4, 4, 1, 5, 6, 2, 3, 2, 3, 4, 1,
 5, 1, 4, 2, 2, 1, 5, 5, 2, 3, 6, 2, 5, 1, 5, 4, 3, 8, 2, 1, 6, 1, 6, 2, 3, 4,
 7, 1, 4, 1, 7, 1, 4, 1, 3, 8, 4, 7, 4, 3, 3, 1, 4, 7, 1, 1, 5, 1, 4, 4, 3, 9,
 1, 3, 8, 8, 4, 6, 4, 4, 2, 1, 2, 3, 4, 3, 10, 6, 1, 3, 3, 6, 5, 6, 1, 2, 4, 8,
 2, 3, 4, 3, 7, 1, 5, 1, 3, 3, 4, 3, 5, 2, 6, 1, 7, 1, 5, 4, 2, 5, 1, 3, 5, 6,
 8, 2, 3, 9, 7, 1, 5, 4, 1, 6, 1, 5, 8, 3, 7, 7, 4, 2, 3, 6, 2, 3, 3, 1, 3, 6,
 3, 3, 8, 6, 7, 4, 1, 4, 1, 5, 6, 4, 1, 8, 1, 5, 2, 3, 4, 1, 3, 3, 3, 5, 6, 4,
 8, 4, 3, 8, 1, 3, 4, 1, 1, 4, 6, 9, 4, 2, 5, 8, 3, 5, 2, 3, 6, 1, 3, 6, 4, 3,
 4, 2, 3, 4, 1, 3, 5, 3, 7, 5, 6, 1, 2, 6, 8, 6, 1, 3, 2, 4, 2, 6, 3, 6, 2, 3,
 5, 5, 2, 4, 1, 4, 5, 3, 6, 4, 1, 4, 9, 1, 3, 6, 1

 6, 1, 4, 1, 2, 3, 4, 3, 1, 4, 1, 3, 3, 1, 1, 4, 2, 2, 5, 1, 5, 1, 6, 1, 3, 1,
 1, 4, 4, 8, 2, 6, 9, 2, 2, 3, 1, 1, 5, 2, 4, 2, 2, 1, 2, 8, 1, 6, 1, 4, 1, 1,
 5, 7, 2, 3, 4, 1, 1, 6, 2, 3, 2, 3, 4, 1, 3, 4, 5, 1, 1, 7, 1, 4, 3, 1, 1, 2,
 1, 4, 1, 4, 2, 4, 6, 1, 3, 5, 3, 4, 2, 2, 2, 1, 3, 1, 4, 2, 4, 4, 2, 3, 2, 1,
 2, 1, 3, 3, 4, 2, 5, 2, 5, 5, 1, 7, 2, 3, 4, 3, 4, 1, 4, 2, 2, 1, 2, 4, 2, 3,
 4, 2, 4, 4, 1, 3, 1, 2, 1, 2, 4, 1, 2, 6, 2, 4, 1, 5, 3, 7, 3, 6, 1, 3, 4, 1,
 2, 4, 2, 5, 5, 3, 7, 2, 2, 1, 2, 1, 2, 1, 6, 3, 2, 1, 1, 1, 7, 9, 2, 3, 1, 1,
 4, 3, 3, 2, 3, 1, 1, 2, 1, 3, 1, 1, 3, 3, 4, 6, 2, 2, 1, 3, 1, 1, 2, 4, 3, 4,
 4, 1, 2, 4, 2, 4, 1, 3, 1, 4, 1, 6, 1, 3, 1, 1, 2, 4, 5, 1, 5, 4, 5, 8, 7, 1,
 3, 3, 1, 4, 3, 1, 5, 4, 1, 3, 4, 2, 7, 3, 2, 1, 2, 1, 2, 1, 2, 3, 4, 1, 1, 4,
 6, 5, 3, 2, 2, 4, 6, 6, 3, 2, 3, 3, 2, 8, 3, 4, 1, 6, 2, 3, 9, 5, 1, 3, 4, 3,
 3, 1, 3, 1, 7, 4, 3, 6, 4, 2, 9, 3, 6, 1, 3, 1, 3, 2, 3, 4, 4, 12, 12, 3, 3, 2,
 3, 9, 7, 2, 3, 8, 4, 1, 1, 3, 10, 5, 6, 2, 5, 5, 

 6, 1, 2, 11, 2, 9, 6, 2, 8, 2, 3, 4, 5, 5, 2, 2, 1, 6, 1, 3, 9, 1, 3, 3, 3, 7,
 2, 4, 10, 7, 1, 4, 2, 4, 1, 1, 7, 2, 3, 1, 4, 4, 3, 2, 2, 4, 1, 5, 4, 5, 5, 6,
 2, 3, 4, 5, 3, 1, 11, 4, 7, 1, 5, 4, 2, 3, 7, 6, 3, 6, 8, 2, 3, 4, 6, 8, 2, 2,
 1, 6, 1, 1, 10, 4, 4, 1, 7, 2, 5, 8, 5, 2, 2, 3, 1, 10, 4, 1, 4, 8, 7, 1, 3, 6,
 9, 2, 3, 5, 11, 1, 4, 7, 8, 3, 3, 6, 2, 4, 6, 4, 2, 9, 4, 5, 1, 3, 9, 4, 3, 6,
 2, 2, 1, 5, 1, 5, 1, 7, 4, 8, 5, 1, 8, 2, 3, 1, 4, 5, 2, 6, 1, 2, 7, 7, 4, 2,
 4, 8, 9, 3, 4, 1, 6, 1, 6, 1, 4, 3, 6, 1, 12, 1, 2, 3, 2, 6, 1, 4, 2, 11, 8, 1,
 2, 3, 4, 3, 5, 6, 3, 1, 2, 2, 3, 3, 3, 5, 1, 1, 5, 3, 3, 3, 2, 3, 4, 1, 3, 1,
 2, 3, 4, 13, 11, 1, 4, 7, 5, 2, 8, 3, 3, 5, 2, 10, 2, 2, 7, 4, 3, 5, 4, 7, 6,
 5, 2, 5, 3, 3, 5, 1, 3, 4, 5, 1, 9, 4, 3, 4, 7, 1, 2, 1, 5, 3, 7, 4, 1, 3, 3,
 3, 7, 1, 3, 2, 6, 1, 6, 1, 8, 1, 3, 8, 5, 8, 3, 5, 4, 2, 3, 5, 1, 3, 5, 6, 2,
 5, 2, 2, 5, 1, 3, 4, 5, 5, 2, 3, 5, 1, 2, 4, 3, 4, 6, 2, 3, 10, 2, 7, 5, 2, 3,
 5, 4, 10, 1, 4, 8, 8, 5, 2, 3, 2, 2, 9, 4,

 1, 5, 1, 3, 2, 2, 1, 5, 6, 4, 2, 1, 7, 4, 1, 7, 5, 1, 5, 2, 3, 3, 2, 8, 4, 8,
 1, 5, 9, 1, 6, 8, 2, 8, 1, 1, 3, 3, 9, 2, 5, 1, 6, 6, 2, 1, 3, 3, 4, 3, 9, 5,
 1, 1, 7, 8, 7, 3, 10, 2, 3, 10, 4, 2, 4, 1, 3, 10, 8, 2, 2, 4, 1, 5, 7, 2, 4,
 7, 1, 5, 3, 4, 2, 3, 3, 2, 2, 4, 8, 5, 1, 2, 2, 6, 4, 1, 4, 5, 2, 10, 8, 3, 3,
 5, 11, 1, 3, 3, 5, 2, 6, 3, 7, 2, 3, 6, 1, 5, 7, 1, 2, 6, 3, 8, 7, 7, 1, 4, 8,
 3, 1, 3, 4, 1, 4, 4, 8, 2, 1, 6, 2, 6, 1, 10, 2, 3, 14, 14, 1, 3, 4, 6, 2, 2,
 3, 4, 9, 3, 2, 2, 9, 2, 1, 7, 1, 2, 2, 6, 6, 3, 3, 3, 4, 4, 3, 8, 1, 3, 1, 1,
 1, 2, 3, 1, 8, 2, 3, 5, 3, 5, 3, 1, 4, 4, 9, 6, 1, 3, 2, 1, 7, 3, 3, 5, 5, 2,
 3, 3, 6, 2, 2, 4, 1, 6, 10, 3, 3, 4, 5, 1, 10, 2, 3, 4, 1, 8, 3, 1, 3, 3, 4, 1,
 5, 3, 1, 3, 3, 4, 6, 1, 4, 4, 1, 3, 6, 2, 7, 1, 3, 5, 4, 4, 1, 4, 10, 1, 6, 2,
 3, 6, 7, 1, 3, 10, 2, 4, 6, 8, 1, 4, 11, 5, 7, 8, 3, 7, 1, 3, 5, 1, 3, 3, 7, 4,
 4, 4, 5, 3, 1, 7, 2, 3, 4, 1, 5, 5, 4, 7, 3, 7, 2, 3, 4, 3, 2, 3, 8, 1, 3, 3,
 6, 7, 2, 3, 4, 4, 5, 7, 1, 4, 9, 1, 1, 9, 1,

 2, 5, 2, 5, 6, 1, 3, 1, 10, 9, 2, 2, 4, 1, 10, 7, 4, 5, 1, 6, 10, 5, 2, 3, 3,
 1, 3, 5, 10, 4, 3, 1, 5, 2, 6, 5, 1, 7, 4, 3, 7, 2, 5, 6, 6, 1, 4, 5, 3, 8, 4,
 7, 4, 9, 6, 1, 3, 4, 10, 8, 2, 14, 2, 6, 3, 7, 9, 1, 6, 7, 8, 2, 4, 2, 9, 1, 3,
 4, 1, 5, 9, 2, 3, 4, 4, 4, 6, 1, 8, 8, 1, 7, 1, 3, 1, 3, 5, 6, 1, 3, 4, 6, 8,
 2, 3, 9, 2, 1, 7, 1, 1, 3, 1, 4, 4, 1, 2, 3, 9, 5, 2, 4, 1, 4, 6, 1, 7, 3, 3,
 7, 1, 8, 1, 9, 1, 7, 1, 3, 13, 7, 4, 5, 2, 5, 1, 4, 1, 9, 1, 8, 3, 9, 2, 3, 9,
 7, 6, 6, 2, 1, 3, 5, 2, 2, 5, 2, 3, 6, 2, 3, 5, 5, 2, 3, 10, 7, 2, 3, 7, 2, 3,
 1, 6, 4, 2, 3, 6, 1, 2, 3, 4, 4, 1, 4, 2, 7, 6, 1, 3, 2, 2, 2, 2, 8, 4, 2, 1,
 5, 3, 4, 7, 8, 3, 9, 1, 3, 12, 8, 2, 3, 5, 1, 2, 2, 9, 2, 5, 5, 1, 3, 5, 2, 4,
 9, 1, 4, 6, 3, 4, 3, 4, 5, 2, 3, 9, 2, 3, 5, 1, 2, 5, 5, 7, 1, 4, 1, 8, 1, 1,
 7, 3, 5, 1, 3, 8, 4, 6, 8, 5, 2, 3, 5, 1, 3, 4, 7, 4, 4, 2, 4, 5, 1, 7, 8, 3,
 3, 5, 2, 3, 4, 1, 6, 4, 1, 8, 2, 3, 4, 2, 3, 5, 6, 5, 4, 13, 1, 2, 2, 2, 4, 13,
 1, 3, 6, 2, 6, 6, 4, 2, 4, 11, 1, 3, 4, 2, 

 3, 3, 6, 2, 5, 6, 7, 2, 10, 4, 4, 2, 3, 6, 1, 3, 4, 12, 3, 9, 2, 5, 3, 12, 5,
 7, 7, 2, 3, 9, 5, 1, 2, 4, 2, 3, 4, 5, 6, 1, 6, 11, 1, 1, 4, 5, 2, 2, 1, 8, 9,
 2, 6, 2, 6, 2, 7, 3, 1, 4, 9, 4, 1, 7, 6, 5, 1, 2, 4, 4, 2, 3, 4, 3, 3, 6, 7,
 1, 3, 3, 2, 5, 2, 4, 2, 3, 10, 2, 6, 1, 3, 3, 4, 2, 5, 3, 2, 1, 6, 3, 5, 2, 4,
 5, 7, 1, 3, 7, 9, 4, 3, 5, 3, 8, 5, 1, 10, 9, 1, 5, 1, 6, 8, 2, 3, 5, 7, 1, 3,
 5, 5, 6, 6, 3, 18, 5, 1, 3, 5, 5, 1, 2, 5, 1, 6, 1, 5, 1, 2, 4, 4, 4, 1, 3, 2,
 3, 3, 4, 10, 2, 3, 10, 1, 3, 12, 3, 4, 9, 2, 9, 1, 9, 5, 8, 4, 1, 6, 1, 8, 6,
 1, 9, 3, 1, 3, 5, 4, 5, 3, 4, 4, 6, 11, 2, 1, 5, 4, 1, 3, 5, 1, 6, 1, 4, 7, 3,
 6, 4, 3, 5, 7, 1, 4, 1, 3, 2, 5, 10, 1, 4, 3, 4, 2, 5, 1, 3, 1, 3, 10, 1, 8, 5,
 3, 5, 2, 4, 1, 8, 1, 10, 4, 1, 4, 2, 7, 11, 5, 6, 4, 3, 4, 4, 2, 5, 1, 7, 3, 2,
 4, 2, 7, 1, 4, 4, 10, 4, 10, 6, 1, 3, 4, 5, 5, 1, 7, 5, 1, 5, 2, 1, 3, 10, 2,
 5, 2, 5, 2, 11, 1, 4, 2, 1, 6, 1, 7, 5, 2, 10, 7, 3, 4, 3, 6, 7, 2, 5, 4, 14,
 2, 3, 5, 1, 3, 3, 6, 1, 5, 1, 6, 1, 4, 2, 

 7, 1, 4, 4, 8, 3, 6, 1, 2, 7, 3, 3, 5, 4, 1, 7, 3, 7, 3, 2, 1, 3, 1, 2, 2, 4,
 5, 4, 7, 1, 3, 5, 8, 2, 3, 9, 1, 2, 3, 9, 4, 3, 8, 1, 3, 1, 3, 3, 4, 3, 4, 3,
 6, 1, 3, 3, 5, 1, 7, 1, 5, 1, 3, 5, 3, 4, 2, 1, 4, 2, 3, 1, 1, 4, 1, 1, 3, 6,
 4, 4, 4, 1, 5, 2, 4, 5, 1, 1, 5, 2, 2, 1, 1, 4, 2, 3, 6, 1, 2, 4, 1, 4, 1, 1,
 4, 3, 4, 2, 3, 2, 8, 3, 8, 1, 3, 1, 3, 3, 3, 4, 2, 6, 3, 3, 2, 3, 2, 4, 4, 1,
 1, 3, 4, 3, 3, 7, 5, 3, 2, 1, 2, 1, 7, 1, 2, 1, 1, 1, 3, 4, 1, 1, 4, 2, 4, 2,
 2, 1, 4, 3, 1, 3, 1, 1, 2, 1, 3, 6, 8, 1, 1, 6, 3, 1, 5, 2, 1, 5, 2, 6, 1, 4,
 2, 1, 6, 1, 2, 1, 4, 2, 1, 4, 2, 3, 5, 1, 1, 4, 2, 3, 4, 4, 2, 6, 3, 11, 1, 6,
 4, 1, 6, 7, 1, 1, 2, 1, 4, 4, 2, 4, 1, 3, 1, 4, 1, 1, 3, 4, 1, 3, 3, 1, 1, 3,
 4, 5, 6, 2, 3, 5, 3, 1, 2, 7, 2, 1, 2, 1, 3, 6, 5, 1, 6, 1, 2, 3, 4, 1, 4, 1,
 3, 1, 4, 2, 1, 1, 3, 8, 6, 2, 3, 3, 3, 3, 2, 4, 6, 1, 2, 3, 4, 5, 2, 3, 1, 5,
 1, 2, 4, 1, 1, 1, 4, 2, 3, 5, 7, 2, 1, 1, 4, 4, 4, 6, 1, 3, 1, 1, 1, 2, 3, 1,
 3, 7, 4, 3, 1, 1, 7, 2, 1, 4, 2, 3, 5, 6, 1, 2, 3,

 2, 3, 4, 3, 2, 4, 3, 7, 4, 3, 5, 4, 3, 4, 5, 5, 3, 5, 1, 3, 3, 5, 6, 9, 3, 4,
 6, 5, 2, 3, 4, 1, 2, 1, 2, 3, 3, 4, 2, 1, 2, 5, 1, 7, 5, 4, 3, 5, 2, 3, 4, 1,
 3, 2, 8, 1, 2, 5, 1, 4, 4, 5, 2, 6, 4, 2, 2, 3, 4, 7, 1, 3, 4, 3, 2, 1, 6, 1,
 6, 1, 4, 14, 7, 7, 11, 7, 5, 3, 8, 1, 2, 3, 6, 10, 2, 3, 7, 2, 3, 7, 9, 1, 3,
 4, 9, 3, 12, 5, 1, 6, 10, 2, 3, 7, 1, 1, 6, 1, 2, 6, 1, 2, 6, 4, 3, 4, 8, 11,
 8, 2, 8, 6, 3, 6, 1, 3, 2, 2, 8, 1, 4, 5, 5, 1, 7, 6, 3, 1, 8, 6, 6, 5, 2, 4,
 2, 3, 4, 1, 6, 1, 3, 4, 5, 1, 3, 5, 1, 1, 4, 3, 3, 2, 3, 4, 1, 3, 2, 3, 3, 3,
 4, 2, 6, 4, 3, 4, 1, 4, 3, 1, 1, 6, 7, 5, 2, 6, 6, 3, 9, 1, 2, 3, 6, 6, 3, 4,
 6, 2, 4, 7, 3, 1, 3, 4, 4, 4, 3, 6, 5, 4, 5, 3, 4, 2, 5, 1, 4, 1, 2, 5, 1, 4,
 4, 2, 3, 1, 3, 3, 3, 9, 4, 3, 6, 4, 2, 1, 3, 7, 3, 8, 2, 3, 4, 1, 3, 7, 8, 1,
 6, 1, 6, 7, 4, 5, 2, 8, 6, 1, 5, 7, 6, 2, 3, 8, 6, 5, 6, 2, 3, 3, 6, 3, 7, 4,
 1, 3, 3, 5, 1, 6, 1, 6, 1, 8, 3, 7, 5, 1, 3, 5, 3, 6, 1, 9, 1, 4, 1, 8, 10, 1,
 3, 3, 3, 13, 1, 3, 1, 11, 1, 5, 7, 1, 4, 3, 2, 3, 

 2, 3, 3, 6, 2, 4, 1, 7, 2, 7, 4, 3, 8, 2, 1, 7, 3, 2, 4, 7, 3, 3, 5, 9, 2, 4,
 3, 8, 4, 3, 4, 1, 4, 1, 3, 2, 2, 6, 4, 3, 6, 5, 1, 3, 11, 2, 3, 7, 2, 3, 4, 1,
 3, 2, 2, 4, 5, 9, 3, 8, 1, 3, 3, 1, 2, 3, 9, 5, 8, 3, 8, 2, 3, 2, 3, 8, 2, 5,
 3, 8, 3, 6, 1, 3, 5, 6, 7, 7, 3, 8, 2, 3, 11, 1, 3, 1, 2, 3, 4, 1, 2, 2, 9, 2,
 3, 4, 1, 3, 1, 4, 8, 1, 7, 1, 3, 8, 1, 3, 10, 1, 3, 1, 3, 4, 4, 9, 7, 3, 4, 3,
 3, 3, 2, 7, 2, 8, 3, 4, 7, 2, 4, 4, 2, 3, 11, 2, 10, 2, 7, 1, 2, 1, 3, 2, 3, 8,
 10, 2, 9, 3, 3, 3, 4, 3, 5, 3, 8, 4, 4, 4, 3, 4, 1, 5, 4, 5, 9, 7, 2, 3, 6, 12,
 1, 7, 5, 7, 3, 7, 2, 6, 1, 3, 1, 4, 1, 5, 4, 2, 4, 3, 4, 1, 5, 5, 4, 1, 3, 3,
 1, 4, 3, 5, 1, 2, 8, 4, 3, 3, 1, 6, 5, 1, 3, 5, 1, 7, 1, 3, 4, 2, 8, 3, 7, 4,
 3, 3, 4, 2, 3, 8, 3, 1, 3, 8, 5, 3, 6, 5, 3, 6, 4, 1, 3, 3, 3, 1, 3, 9, 10, 5,
 6, 3, 8, 4, 1, 4, 2, 4, 4, 6, 1, 4, 2, 4, 3, 1, 7, 3, 6, 2, 5, 6, 1, 3, 3, 5,
 3, 5, 1, 3, 1, 3, 1, 2, 2, 4, 7, 2, 3, 4, 4, 4, 1, 3, 3, 1, 2, 7, 1, 5, 6, 2,
 1, 6, 1, 2, 6, 1, 4, 3, 4, 4, 2, 6, 1, 3, 3

 1, 9, 2, 1, 6, 3, 7, 5, 1, 5, 3, 1, 1, 1, 3, 1, 1, 3, 3, 5, 7, 5, 1, 4, 1, 4,
 1, 3, 5, 4, 3, 7, 1, 6, 2, 2, 4, 4, 5, 4, 2, 1, 4, 3, 3, 1, 5, 9, 1, 3, 2, 3,
 8, 1, 3, 3, 5, 7, 4, 5, 5, 1, 3, 3, 7, 6, 7, 1, 4, 1, 3, 5, 3, 3, 1, 3, 5, 8,
 2, 3, 3, 1, 4, 3, 7, 7, 2, 4, 1, 3, 2, 4, 13, 2, 3, 1, 4, 3, 8, 5, 4, 3, 5, 2,
 3, 4, 1, 2, 3, 4, 7, 2, 3, 5, 2, 2, 6, 1, 2, 3, 6, 2, 2, 5, 4, 2, 3, 4, 6, 3,
 4, 1, 4, 7, 4, 1, 3, 2, 3, 4, 4, 4, 6, 2, 5, 3, 3, 6, 1, 2, 7, 4, 1, 4, 1, 4,
 4, 3, 2, 3, 7, 5, 2, 3, 6, 1, 4, 2, 3, 9, 5, 6, 3, 4, 8, 1, 3, 6, 6, 4, 3, 7,
 4, 1, 4, 3, 5, 3, 7, 6, 5, 2, 7, 1, 3, 5, 7, 8, 1, 3, 4, 1, 7, 3, 3, 5, 3, 1,
 3, 8, 3, 5, 3, 1, 4, 3, 5, 1, 4, 6, 1, 5, 3, 2, 3, 6, 3, 1, 1, 3, 1, 3, 1, 6,
 1, 3, 2, 5, 5, 8, 3, 2, 3, 3, 1, 3, 3, 2, 3, 4, 5, 3, 6, 1, 1, 1, 2, 3, 1, 3,
 1, 3, 4, 5, 3, 3, 1, 5, 3, 1, 2, 4, 3, 3, 5, 5, 3, 3, 4, 3, 2, 1, 3, 5, 5, 2,
 4, 3, 10, 1, 8, 3, 5, 3, 7, 3, 4, 1, 3, 6, 1, 6, 5, 5, 2, 1, 2, 2, 2, 3, 4, 1,
 9, 1, 1, 5, 1, 3, 3, 1, 2, 3, 4, 1, 4, 6, 2, 1, 3

 2, 7, 1, 2, 1, 3, 3, 1, 2, 4, 5, 1, 1, 1, 3, 7, 2, 6, 1, 2, 1, 3, 3, 1, 5, 1,
 6, 6, 1, 3, 1, 3, 3, 1, 6, 1, 8, 2, 8, 7, 2, 3, 6, 2, 6, 1, 2, 1, 3, 7, 2, 4,
 1, 4, 11, 6, 3, 10, 3, 2, 3, 7, 1, 3, 2, 2, 3, 7, 4, 1, 7, 1, 5, 3, 6, 2, 7, 4,
 5, 5, 1, 7, 1, 6, 1, 3, 5, 2, 4, 3, 3, 4, 5, 1, 2, 3, 4, 1, 1, 7, 5, 4, 2, 3,
 8, 9, 2, 3, 4, 4, 1, 4, 1, 3, 5, 1, 6, 5, 4, 8, 8, 4, 12, 9, 2, 2, 1, 7, 2, 1,
 14, 3, 8, 11, 1, 3, 1, 11, 1, 1, 8, 13, 6, 2, 2, 3, 1, 8, 3, 8, 8, 2, 5, 1, 4,
 1, 3, 2, 8, 2, 5, 3, 5, 10, 2, 4, 7, 9, 3, 11, 5, 7, 1, 2, 3, 5, 5, 1, 3, 6, 2,
 6, 1, 3, 5, 3, 6, 6, 8, 1, 5, 7, 1, 1, 6, 2, 1, 6, 1, 2, 5, 9, 8, 7, 1, 5, 8,
 5, 4, 8, 2, 3, 6, 2, 5, 8, 1, 3, 10, 2, 5, 8, 5, 4, 2, 5, 6, 5, 4, 1, 2, 3, 4,
 6, 2, 4, 9, 7, 5, 3, 9, 7, 6, 4, 5, 2, 7, 2, 5, 3, 1, 3, 8, 2, 4, 1, 3, 1, 2,
 4, 5, 1, 2, 1, 3, 7, 1, 3, 1, 2, 5, 5, 9, 5, 4, 2, 10, 2, 3, 5, 6, 2, 4, 7, 1,
 2, 4, 3, 5, 6, 2, 3, 2, 5, 5, 8, 1, 9, 3, 5, 6, 2, 3, 4, 3, 11, 3, 1, 3, 3, 4,
 6, 5, 6, 1, 3, 1, 3, 9, 2, 3, 2, 4, 2, 5, 

 4, 1, 4, 4, 1, 6, 1, 3, 7, 3, 5, 2, 5, 3, 4, 2, 5, 1, 5, 1, 3, 6, 2, 1, 9, 4,
 3, 10, 4, 2, 4, 4, 4, 5, 1, 3, 6, 6, 7, 1, 4, 2, 1, 6, 5, 8, 2, 3, 4, 3, 6, 1,
 2, 5, 3, 4, 2, 7, 9, 5, 1, 3, 3, 5, 2, 3, 6, 2, 1, 4, 2, 6, 4, 2, 3, 6, 1, 5,
 3, 5, 4, 5, 3, 5, 4, 3, 3, 1, 3, 5, 3, 6, 4, 4, 5, 4, 1, 2, 1, 4, 4, 1, 5, 1,
 1, 3, 2, 2, 10, 6, 3, 1, 7, 1, 5, 5, 4, 3, 6, 4, 3, 6, 4, 9, 1, 5, 2, 4, 1, 3,
 4, 2, 5, 6, 2, 7, 2, 1, 3, 4, 1, 5, 1, 1, 4, 1, 4, 3, 1, 4, 4, 3, 1, 2, 4, 2,
 4, 4, 5, 1, 7, 5, 6, 1, 3, 2, 1, 5, 10, 6, 1, 4, 1, 7, 3, 8, 1, 1, 2, 7, 1, 4,
 4, 4, 6, 1, 3, 4, 2, 3, 5, 2, 6, 1, 4, 3, 7, 5, 3, 4, 5, 2, 3, 5, 2, 3, 4, 2,
 3, 6, 2, 2, 4, 5, 6, 2, 3, 5, 1, 5, 3, 1, 6, 1, 3, 2, 3, 2, 1, 11, 1, 3, 1, 8,
 1, 3, 4, 8, 7, 5, 1, 7, 3, 4, 3, 5, 1, 1, 9, 8, 3, 4, 2, 6, 1, 8, 1, 4, 1, 2,
 4, 9, 5, 11, 4, 4, 2, 4, 3, 4, 1, 6, 2, 3, 10, 1, 2, 3, 2, 5, 8, 1, 7, 1, 2, 3,
 4, 6, 2, 3, 11, 1, 2, 3, 3, 11, 4, 3, 8, 4, 2, 3, 5, 1, 3, 4, 13, 2, 4, 1, 4,
 3, 4, 3, 3, 6, 2, 3, 7, 3, 8, 2, 9, 2, 5, 5, 

 1, 9, 1, 3, 2, 4, 3, 4, 4, 1, 2, 1, 2, 1, 2, 4, 5, 4, 1, 2, 1, 3, 1, 2, 4, 5,
 5, 2, 6, 1, 6, 1, 2, 4, 1, 1, 4, 1, 4, 4, 1, 1, 4, 2, 4, 9, 2, 4, 8, 5, 4, 3,
 4, 1, 3, 1, 4, 1, 1, 10, 5, 5, 3, 3, 10, 1, 1, 4, 2, 4, 1, 1, 2, 5, 1, 2, 4, 2,
 7, 3, 6, 1, 5, 1, 3, 1, 3, 1, 5, 1, 2, 1, 1, 1, 11, 1, 2, 8, 3, 6, 4, 1, 2, 5,
 1, 6, 3, 5, 5, 1, 3, 1, 3, 1, 2, 4, 4, 3, 5, 4, 3, 4, 4, 2, 4, 4, 1, 2, 1, 1,
 1, 2, 2, 4, 1, 1, 2, 4, 2, 3, 1, 4, 7, 5, 1, 4, 4, 3, 5, 1, 10, 1, 3, 4, 3, 1,
 4, 3, 1, 6, 4, 4, 1, 10, 1, 3, 8, 4, 4, 4, 1, 3, 4, 1, 1, 4, 2, 4, 4, 8, 6, 2,
 4, 1, 3, 2, 4, 1, 1, 8, 1, 3, 3, 4, 5, 1, 4, 4, 2, 3, 9, 3, 5, 4, 3, 8, 2, 1,
 9, 1, 4, 2, 1, 1, 7, 4, 4, 6, 1, 4, 3, 2, 5, 1, 4, 2, 5, 8, 1, 6, 1, 5, 1, 1,
 1, 6, 6, 4, 1, 8, 4, 1, 3, 1, 4, 1, 5, 10, 2, 5, 2, 1, 1, 1, 2, 5, 1, 8, 3, 5,
 1, 9, 7, 1, 8, 1, 5, 4, 4, 2, 3, 5, 1, 4, 1, 5, 8, 5, 3, 6, 6, 1, 4, 1, 4, 4,
 5, 2, 1, 2, 1, 2, 4, 2, 3, 5, 1, 4, 1, 4, 5, 4, 7, 3, 5, 1, 2, 5, 2, 3, 1, 5,
 8, 1, 3, 5, 1, 7, 2, 2, 4, 2, 4, 6, 1, 3, 3, 

 5, 7, 2, 3, 4, 1, 7, 2, 3, 1, 3, 6, 4, 8, 1, 3, 3, 7, 4, 7, 1, 7, 3, 8, 2, 3,
 6, 1, 4, 9, 1, 2, 3, 5, 4, 3, 5, 4, 6, 4, 3, 4, 4, 9, 5, 1, 3, 4, 3, 4, 4, 2,
 7, 3, 1, 2, 1, 2, 5, 4, 7, 1, 4, 5, 3, 6, 3, 6, 1, 10, 3, 8, 2, 3, 4, 6, 1, 4,
 5, 1, 3, 4, 4, 5, 8, 7, 5, 8, 5, 4, 3, 8, 5, 1, 2, 4, 3, 4, 2, 1, 3, 2, 7, 1,
 3, 8, 1, 3, 5, 3, 6, 2, 3, 9, 9, 1, 4, 3, 6, 6, 4, 3, 4, 6, 6, 1, 5, 4, 10, 5,
 2, 7, 4, 1, 4, 6, 2, 3, 6, 8, 1, 6, 7, 3, 10, 3, 5, 4, 3, 5, 1, 6, 5, 6, 1, 3,
 1, 2, 5, 1, 3, 5, 1, 3, 4, 1, 3, 6, 2, 6, 1, 3, 4, 7, 2, 5, 1, 4, 7, 3, 1, 4,
 2, 4, 3, 5, 5, 4, 4, 2, 3, 5, 1, 3, 3, 6, 4, 6, 5, 4, 1, 2, 2, 3, 4, 8, 4, 5,
 9, 1, 3, 6, 3, 6, 2, 3, 3, 4, 2, 3, 1, 9, 8, 1, 3, 1, 5, 3, 3, 4, 2, 1, 10, 7,
 4, 3, 9, 2, 3, 4, 1, 4, 3, 2, 4, 3, 8, 3, 3, 5, 4, 4, 1, 3, 1, 2, 4, 4, 2, 3,
 4, 2, 4, 5, 9, 1, 3, 4, 5, 1, 6, 1, 8, 1, 3, 6, 4, 1, 5, 1, 5, 7, 3, 4, 2, 3,
 7, 3, 1, 2, 2, 4, 1, 4, 2, 4, 8, 2, 7, 1, 9, 3, 7, 1, 7, 3, 6, 2, 4, 2, 4, 1,
 4, 3, 3, 2, 2, 7, 1, 3, 6, 3, 4, 3, 6, 3, 4, 1,

 4, 3, 4, 3, 5, 1, 10, 2, 1, 3, 1, 3, 1, 1, 5, 1, 3, 1, 3, 5, 3, 5, 3, 3, 2, 1,
 4, 2, 4, 1, 4, 2, 5, 4, 1, 12, 9, 1, 3, 7, 4, 4, 5, 2, 5, 1, 1, 9, 1, 1, 4, 2,
 4, 2, 4, 2, 3, 2, 7, 3, 1, 4, 4, 4, 1, 10, 2, 8, 3, 4, 6, 3, 6, 4, 1, 1, 2, 4,
 5, 1, 2, 1, 10, 1, 4, 6, 5, 1, 3, 2, 4, 4, 3, 5, 1, 3, 4, 3, 8, 6, 7, 2, 4, 1,
 6, 1, 3, 1, 2, 4, 7, 4, 4, 2, 1, 2, 1, 4, 2, 3, 3, 1, 3, 1, 4, 1, 3, 3, 5, 3,
 5, 3, 3, 2, 1, 3, 1, 10, 1, 2, 2, 3, 3, 1, 3, 1, 3, 1, 2, 2, 12, 1, 3, 6, 4, 4,
 3, 1, 2, 2, 4, 2, 2, 5, 1, 2, 3, 6, 4, 4, 3, 4, 2, 2, 5, 2, 4, 2, 3, 8, 1, 3,
 1, 4, 4, 5, 1, 1, 4, 1, 2, 4, 5, 3, 5, 2, 1, 3, 1, 4, 2, 4, 2, 8, 3, 8, 3, 1,
 2, 2, 2, 4, 4, 2, 4, 5, 1, 1, 1, 1, 3, 1, 4, 1, 7, 4, 1, 3, 4, 1, 8, 4, 2, 6,
 5, 3, 4, 1, 5, 1, 9, 4, 4, 5, 2, 1, 5, 4, 3, 3, 3, 1, 4, 1, 5, 1, 1, 3, 5, 2,
 8, 3, 5, 4, 3, 5, 1, 1, 4, 2, 1, 10, 1, 5, 3, 3, 8, 4, 1, 6, 2, 3, 5, 5, 2, 6,
 6, 2, 1, 5, 1, 4, 6, 1, 3, 1, 3, 1, 7, 4, 1, 4, 4, 4, 1, 4, 1, 3, 4, 3, 11, 5,
 2, 11, 4, 4, 2, 1, 1, 4, 2, 1, 3, 3, 1, 5, 

 5, 2, 6, 2, 4, 5, 1, 1, 7, 3, 3, 7, 2, 3, 1, 1, 4, 1, 3, 4, 4, 3, 5, 2, 3, 4,
 1, 2, 2, 3, 3, 5, 3, 2, 4, 2, 3, 9, 1, 5, 4, 1, 4, 8, 2, 7, 2, 2, 3, 4, 1, 1,
 2, 6, 4, 1, 5, 2, 7, 3, 2, 4, 1, 3, 5, 4, 1, 1, 7, 1, 3, 3, 5, 13, 1, 2, 3, 3,
 2, 4, 1, 5, 5, 7, 2, 1, 3, 4, 2, 2, 5, 2, 3, 3, 1, 3, 2, 8, 4, 3, 4, 2, 3, 4,
 1, 1, 4, 1, 7, 2, 4, 1, 4, 6, 5, 4, 6, 8, 2, 4, 1, 4, 1, 5, 3, 7, 4, 2, 2, 3,
 7, 3, 3, 4, 5, 3, 4, 8, 4, 3, 4, 3, 4, 4, 1, 1, 8, 7, 4, 6, 5, 3, 4, 1, 3, 1,
 10, 7, 4, 6, 5, 3, 7, 2, 5, 6, 4, 3, 4, 1, 10, 6, 1, 1, 4, 1, 10, 2, 4, 3, 6,
 4, 1, 5, 1, 9, 2, 3, 6, 2, 5, 9, 1, 3, 6, 9, 8, 2, 3, 10, 1, 4, 4, 1, 10, 6, 3,
 4, 1, 7, 3, 3, 2, 3, 6, 1, 2, 4, 3, 6, 2, 2, 2, 2, 7, 3, 5, 5, 5, 1, 3, 2, 3,
 4, 4, 1, 7, 4, 3, 4, 1, 3, 5, 4, 1, 6, 8, 5, 7, 6, 4, 3, 4, 1, 6, 4, 1, 6, 2,
 3, 5, 7, 11, 2, 3, 10, 5, 1, 8, 7, 3, 6, 1, 5, 1, 8, 7, 3, 6, 1, 1, 10, 3, 1,
 3, 4, 1, 1, 4, 1, 2, 2, 4, 8, 5, 3, 4, 1, 1, 1, 4, 3, 6, 3, 5, 4, 4, 3, 7, 2,
 3, 5, 1, 4, 2, 3, 9, 7, 4, 1, 3, 1, 4, 4, 2, 4, 

 3, 8, 2, 3, 5, 1, 4, 1, 2, 13, 6, 6, 2, 1, 6, 1, 3, 5, 4, 2, 6, 4, 6, 4, 3, 3,
 4, 2, 3, 7, 8, 1, 3, 7, 6, 7, 4, 3, 3, 3, 1, 1, 6, 4, 3, 1, 4, 3, 1, 4, 5, 6,
 8, 1, 2, 5, 2, 5, 6, 1, 3, 8, 17, 2, 3, 5, 4, 1, 7, 7, 7, 2, 5, 5, 1, 6, 2, 3,
 6, 5, 2, 3, 3, 4, 2, 3, 8, 1, 3, 6, 7, 2, 3, 4, 4, 9, 8, 1, 2, 4, 5, 7, 7, 3,
 7, 8, 1, 8, 3, 2, 6, 1, 7, 3, 1, 3, 3, 2, 8, 1, 1, 5, 1, 2, 4, 7, 4, 2, 3, 3,
 7, 1, 5, 9, 7, 2, 4, 7, 1, 3, 2, 5, 4, 3, 7, 2, 4, 14, 4, 3, 3, 7, 2, 6, 4, 2,
 3, 6, 1, 1, 2, 5, 4, 4, 2, 4, 2, 2, 8, 1, 1, 3, 1, 5, 1, 1, 3, 4, 6, 4, 2, 7,
 2, 2, 3, 4, 3, 4, 3, 7, 5, 1, 3, 2, 5, 2, 2, 4, 2, 1, 7, 1, 4, 1, 1, 3, 4, 7,
 2, 4, 6, 6, 1, 7, 1, 4, 1, 6, 5, 2, 3, 1, 6, 1, 1, 4, 3, 1, 3, 1, 3, 1, 3, 6,
 2, 3, 3, 3, 4, 1, 1, 3, 4, 1, 4, 5, 2, 3, 4, 1, 1, 1, 2, 4, 6, 2, 4, 1, 1, 1,
 2, 4, 3, 6, 6, 5, 1, 5, 3, 1, 4, 2, 6, 1, 4, 4, 2, 3, 1, 3, 3, 4, 1, 4, 3, 5,
 3, 4, 3, 4, 1, 1, 4, 3, 4, 4, 1, 2, 1, 1, 1, 4, 4, 4, 1, 1, 4, 4, 3, 4, 8, 4,
 3, 3, 4, 4, 2, 7, 2, 2, 3, 3, 4, 3, 1, 4, 3, 2, 

 1, 2, 4, 4, 1, 4, 4, 3, 7, 2, 4, 5, 1, 2, 5, 2, 1, 7, 1, 3, 4, 8, 2, 1, 3, 2,
 1, 13, 2, 4, 2, 3, 6, 1, 3, 2, 3, 10, 2, 2, 4, 7, 8, 1, 4, 8, 4, 3, 4, 4, 3, 6,
 2, 5, 4, 7, 4, 3, 5, 3, 5, 6, 1, 3, 5, 3, 3, 7, 4, 3, 1, 3, 4, 4, 1, 4, 3, 3,
 2, 2, 8, 1, 8, 3, 8, 2, 3, 2, 2, 1, 3, 2, 2, 4, 2, 10, 9, 5, 2, 5, 1, 3, 2, 2,
 3, 8, 9, 2, 4, 6, 3, 11, 6, 2, 3, 6, 1, 3, 5, 4, 9, 2, 3, 4, 3, 4, 3, 8, 9, 4,
 3, 5, 3, 1, 8, 4, 5, 5, 2, 2, 1, 9, 1, 4, 4, 5, 2, 5, 3, 4, 1, 1, 5, 10, 4, 1,
 4, 1, 4, 2, 7, 7, 5, 3, 9, 8, 6, 1, 4, 10, 3, 6, 1, 1, 1, 4, 1, 4, 2, 1, 6, 2,
 5, 8, 1, 8, 1, 1, 6, 5, 1, 4, 1, 1, 6, 3, 2, 4, 5, 1, 1, 2, 4, 4, 1, 4, 6, 3,
 2, 4, 5, 1, 1, 3, 9, 4, 3, 7, 2, 6, 2, 1, 5, 2, 2, 1, 5, 1, 1, 7, 4, 3, 2, 2,
 4, 8, 1, 5, 1, 10, 5, 1, 1, 3, 2, 1, 9, 1, 2, 3, 4, 2, 2, 2, 2, 3, 6, 1, 1, 4,
 1, 4, 4, 4, 2, 1, 3, 2, 2, 1, 3, 5, 1, 4, 4, 3, 3, 1, 3, 2, 6, 3, 4, 2, 4, 1,
 6, 1, 1, 3, 2, 1, 3, 5, 1, 5, 3, 5, 1, 3, 2, 6, 3, 4, 2, 4, 1, 8, 1, 4, 1, 1,
 3, 2, 1, 3, 5, 1, 4, 3, 4, 3, 4, 4, 5, 1, 3,

 5, 3, 9, 1, 4, 2, 4, 3, 3, 3, 1, 6, 5, 3, 1, 4, 4, 4, 1, 4, 1, 3, 4, 1, 3, 8,
 1, 6, 4, 1, 5, 1, 4, 4, 3, 4, 3, 1, 2, 1, 4, 4, 1, 4, 2, 5, 1, 8, 1, 3, 3, 1,
 1, 5, 5, 4, 3, 7, 1, 1, 4, 2, 5, 1, 5, 4, 4, 1, 3, 3, 4, 4, 3, 5, 1, 5, 4, 3,
 7, 3, 4, 1, 5, 5, 1, 1, 4, 5, 2, 2, 1, 6, 1, 6, 1, 1, 2, 6, 4, 1, 7, 4, 3, 4,
 4, 2, 1, 3, 4, 1, 1, 2, 4, 1, 6, 4, 5, 7, 5, 1, 3, 4, 5, 7, 1, 3, 7, 4, 2, 4,
 2, 2, 7, 3, 3, 3, 1, 3, 3, 4, 2, 3, 6, 4, 3, 5, 1, 5, 2, 4, 3, 4, 1, 11, 6, 1,
 4, 3, 3, 5, 7, 6, 5, 2, 6, 2, 4, 6, 3, 3, 4, 2, 5, 1, 4, 1, 2, 4, 3, 3, 4, 8,
 5, 1, 3, 1, 4, 2, 5, 7, 4, 2, 3, 4, 8, 3, 9, 5, 3, 10, 5, 3, 4, 9, 1, 3, 4, 1,
 6, 3, 4, 6, 5, 2, 3, 4, 3, 9, 3, 8, 11, 4, 4, 3, 3, 3, 4, 3, 3, 10, 8, 4, 7, 5,
 13, 9, 2, 4, 3, 8, 12, 4, 5, 3, 5, 5, 4, 3, 7, 1, 7, 7, 2, 6, 7, 1, 3, 2, 3, 4,
 5, 6, 1, 9, 2, 3, 3, 3, 1, 1, 4, 1, 5, 2, 3, 6, 10, 3, 6, 5, 2, 3, 1, 2, 3, 9,
 5, 4, 1, 5, 7, 3, 8, 1, 6, 1, 3, 6, 1, 5, 8, 3, 8, 1, 6, 1, 7, 4, 1, 2, 4, 1,
 1, 7, 3, 7, 7, 4, 4, 3, 8, 8, 2, 3, 6, 4, 1,

 8, 7, 1, 3, 3, 4, 2, 2, 9, 1, 3, 5, 6, 9, 5, 9, 3, 12, 10, 2, 8, 1, 2, 3, 5, 2,
 1, 4, 5, 2, 3, 3, 8, 3, 10, 5, 1, 6, 2, 3, 1, 3, 10, 2, 3, 7, 6, 8, 9, 2, 1, 3,
 8, 8, 1, 4, 5, 2, 4, 5, 4, 1, 3, 1, 2, 3, 5, 2, 1, 4, 1, 5, 6, 1, 1, 3, 3, 4,
 1, 5, 1, 4, 3, 5, 2, 10, 4, 3, 5, 2, 5, 7, 10, 2, 3, 10, 4, 1, 6, 6, 3, 8, 4,
 4, 8, 5, 4, 4, 6, 2, 6, 4, 3, 7, 1, 2, 15, 3, 4, 6, 4, 5, 7, 2, 5, 5, 1, 1, 4,
 1, 1, 2, 3, 2, 5, 4, 1, 10, 3, 4, 2, 3, 3, 1, 1, 7, 1, 7, 3, 2, 3, 3, 5, 7, 4,
 2, 5, 1, 1, 1, 1, 7, 5, 3, 9, 1, 4, 3, 5, 1, 3, 2, 4, 5, 2, 3, 4, 5, 3, 1, 2,
 3, 4, 5, 3, 4, 1, 4, 1, 8, 3, 7, 3, 2, 3, 4, 5, 1, 4, 4, 1, 3, 2, 1, 3, 7, 4,
 4, 8, 3, 7, 1, 4, 1, 5, 3, 3, 6, 2, 3, 3, 5, 1, 3, 5, 7, 5, 7, 3, 10, 7, 7, 3,
 4, 1, 3, 1, 4, 3, 3, 1, 7, 1, 4, 1, 2, 6, 1, 4, 1, 4, 3, 2, 3, 3, 1, 1, 5, 3,
 5, 3, 5, 3, 5, 1, 1, 4, 4, 1, 4, 1, 4, 1, 2, 2, 4, 4, 2, 5, 2, 3, 3, 5, 6, 14,
 5, 2, 2, 3, 5, 4, 1, 4, 1, 3, 3, 2, 4, 6, 2, 3, 7, 2, 6, 3, 6, 5, 4, 3, 2, 4,
 4, 8, 1, 4, 3, 3, 6, 3, 5, 5, 1, 4, 4, 5, 3

 2, 2, 3, 7, 8, 4, 4, 2, 8, 2, 3, 9, 4, 1, 3, 2, 3, 6, 4, 9, 2, 3, 6, 7, 2, 2,
 10, 5, 2, 3, 5, 4, 1, 4, 1, 4, 7, 2, 5, 5, 1, 3, 4, 3, 4, 2, 7, 4, 2, 1, 4, 1,
 1, 5, 1, 3, 7, 3, 2, 2, 1, 3, 2, 5, 2, 3, 4, 5, 4, 5, 4, 2, 4, 1, 4, 1, 2, 4,
 2, 4, 2, 4, 1, 4, 2, 1, 6, 1, 5, 3, 9, 4, 5, 12, 4, 4, 5, 9, 2, 3, 4, 1, 9, 2,
 3, 8, 5, 2, 3, 5, 1, 3, 3, 2, 10, 1, 2, 4, 5, 5, 4, 5, 4, 4, 4, 3, 7, 5, 1, 4,
 4, 5, 7, 4, 2, 4, 6, 2, 8, 2, 9, 1, 3, 2, 6, 4, 9, 2, 3, 10, 2, 3, 3, 1, 3, 4,
 3, 3, 3, 4, 2, 3, 8, 1, 1, 6, 1, 3, 2, 4, 7, 5, 2, 3, 8, 2, 9, 7, 2, 3, 5, 4,
 7, 3, 2, 2, 7, 6, 1, 5, 7, 3, 1, 7, 4, 3, 4, 1, 9, 4, 5, 10, 1, 3, 4, 10, 7, 7,
 8, 3, 5, 2, 6, 1, 3, 4, 4, 2, 1, 7, 4, 1, 6, 2, 3, 3, 2, 4, 1, 6, 1, 4, 1, 7,
 1, 7, 1, 6, 1, 5, 3, 5, 2, 3, 5, 5, 1, 1, 2, 4, 3, 4, 2, 5, 5, 1, 1, 3, 4, 5,
 6, 2, 3, 4, 6, 2, 3, 1, 1, 4, 3, 7, 2, 3, 5, 1, 4, 4, 3, 1, 3, 3, 5, 7, 3, 2,
 1, 2, 1, 4, 3, 4, 2, 2, 1, 3, 6, 2, 1, 5, 1, 4, 1, 2, 3, 3, 3, 3, 2, 4, 1, 4,
 1, 2, 5, 1, 1, 1, 4, 1, 2, 1, 2, 1, 2, 2, 3, 

 2, 3, 7, 9, 1, 3, 6, 4, 6, 4, 4, 3, 6, 2, 3, 9, 3, 5, 2, 3, 3, 5, 4, 1, 1, 7,
 3, 4, 2, 3, 6, 3, 3, 4, 3, 2, 8, 8, 2, 7, 1, 2, 2, 6, 3, 3, 1, 6, 4, 1, 5, 4,
 3, 4, 1, 1, 3, 4, 2, 1, 3, 7, 2, 5, 4, 4, 6, 2, 1, 3, 4, 3, 6, 3, 2, 5, 1, 1,
 8, 2, 4, 1, 7, 2, 3, 2, 7, 3, 2, 1, 5, 2, 2, 4, 2, 2, 2, 5, 4, 1, 9, 2, 3, 6,
 6, 1, 1, 2, 6, 4, 3, 4, 2, 1, 3, 1, 3, 1, 3, 2, 3, 11, 8, 3, 4, 1, 1, 7, 1, 3,
 4, 4, 4, 5, 2, 4, 1, 1, 5, 4, 3, 6, 1, 4, 1, 6, 3, 2, 4, 1, 4, 6, 3, 4, 4, 7,
 4, 4, 8, 1, 3, 2, 4, 1, 4, 5, 6, 3, 3, 7, 3, 1, 4, 8, 3, 5, 5, 1, 3, 9, 4, 3,
 5, 4, 1, 5, 5, 7, 4, 2, 2, 2, 3, 7, 2, 4, 3, 6, 2, 1, 1, 8, 2, 8, 1, 4, 4, 1,
 4, 5, 4, 1, 4, 4, 1, 2, 4, 5, 3, 5, 7, 2, 1, 5, 3, 4, 3, 4, 1, 1, 3, 1, 2, 1,
 1, 3, 3, 3, 2, 5, 3, 1, 2, 1, 8, 1, 3, 5, 5, 1, 5, 4, 4, 8, 4, 1, 3, 1, 4, 1,
 2, 1, 2, 1, 2, 7, 4, 1, 3, 1, 4, 4, 1, 3, 3, 2, 4, 3, 4, 4, 5, 4, 5, 4, 2, 3,
 3, 1, 5, 1, 1, 4, 3, 4, 1, 2, 3, 1, 3, 5, 1, 3, 3, 7, 2, 5, 1, 3, 1, 3, 3, 7,
 1, 4, 5, 1, 5, 3, 2, 1, 1, 3, 1, 4, 3, 4, 1, 5, 2,

 4, 2, 4, 4, 7, 3, 1, 11, 7, 3, 3, 8, 6, 7, 1, 2, 2, 3, 7, 4, 7, 1, 4, 3, 9, 5,
 3, 2, 5, 1, 5, 5, 2, 11, 1, 2, 8, 1, 2, 4, 1, 2, 5, 4, 5, 7, 1, 3, 5, 11, 5,
 10, 5, 7, 8, 2, 3, 3, 1, 3, 3, 1, 10, 7, 6, 3, 4, 4, 6, 1, 3, 3, 3, 4, 1, 3, 4,
 8, 3, 1, 3, 5, 5, 1, 4, 1, 3, 8, 1, 7, 3, 9, 1, 3, 5, 3, 5, 2, 3, 6, 1, 7, 3,
 8, 4, 2, 3, 6, 3, 2, 5, 1, 3, 2, 3, 4, 9, 13, 6, 10, 4, 3, 6, 4, 2, 3, 6, 2, 3,
 8, 1, 7, 1, 3, 6, 2, 3, 5, 4, 3, 7, 1, 8, 4, 4, 9, 5, 2, 3, 4, 1, 5, 6, 9, 1,
 4, 8, 11, 1, 4, 3, 10, 7, 4, 6, 7, 3, 2, 4, 7, 1, 1, 3, 3, 3, 1, 6, 7, 10, 8,
 2, 3, 4, 3, 1, 2, 3, 3, 2, 7, 3, 4, 6, 3, 3, 6, 2, 3, 9, 4, 1, 1, 3, 1, 4, 4,
 10, 3, 8, 3, 4, 1, 5, 9, 4, 3, 7, 4, 5, 3, 5, 1, 7, 2, 3, 4, 1, 3, 11, 8, 5, 1,
 4, 4, 3, 7, 2, 9, 3, 5, 10, 4, 7, 3, 6, 3, 3, 4, 1, 3, 14, 7, 4, 8, 6, 1, 3, 9,
 6, 7, 1, 2, 3, 5, 1, 4, 1, 4, 3, 6, 6, 5, 1, 3, 2, 3, 4, 2, 6, 1, 4, 11, 7, 1,
 3, 3, 8, 4, 6, 7, 1, 3, 3, 5, 7, 4, 6, 2, 3, 4, 2, 4, 1, 4, 4, 4, 3, 4, 1, 5,
 1, 4, 5, 1, 5, 3, 4, 2, 4, 4, 3, 3, 7, 1, 

In [63]:
fdist = FreqDist(len(w) for w in text1)
print(fdist)

<FreqDist with 19 samples and 260819 outcomes>


In [64]:
fdist

FreqDist({3: 50223, 1: 47933, 4: 42345, 2: 38513, 5: 26597, 6: 17111, 7: 14399, 8: 9966, 9: 6428, 10: 3528, ...})

In [65]:
pp.pprint(fdist)

{1: 47933,
 2: 38513,
 3: 50223,
 4: 42345,
 5: 26597,
 6: 17111,
 7: 14399,
 8: 9966,
 9: 6428,
 10: 3528,
 11: 1873,
 12: 1053,
 13: 567,
 14: 177,
 15: 70,
 16: 22,
 17: 12,
 18: 1,
 20: 1}


In [66]:
fdist.most_common()

[(3, 50223),
 (1, 47933),
 (4, 42345),
 (2, 38513),
 (5, 26597),
 (6, 17111),
 (7, 14399),
 (8, 9966),
 (9, 6428),
 (10, 3528),
 (11, 1873),
 (12, 1053),
 (13, 567),
 (14, 177),
 (15, 70),
 (16, 22),
 (17, 12),
 (18, 1),
 (20, 1)]

In [67]:
fdist.max()

3

In [68]:
fdist.freq(3)

0.19255882431878046

### Functions Defined for NLTK's Frequency Distributions
|Example|Description|
|:-------|:-----------|
|fdist = FreqDist(samples)	|create a frequency distribution containing the given samples|
|fdist[sample] += 1	        |increment the count for this sample|
|fdist['monstrous']	        |count of the number of times a given sample occurred|
|fdist.freq('monstrous')	|frequency of a given sample|
|fdist.N()	                |total number of samples|
|fdist.most_common(n)	    |the n most common samples and their frequencies|
|for sample in fdist:	    |iterate over the samples|
|fdist.max()	            |sample with the greatest count|
|fdist.tabulate()	        |tabulate the frequency distribution|
|fdist.plot()	            |graphical plot of the frequency distribution|
|fdist.plot(cumulative=True)|cumulative plot of the frequency distribution|
|fdist1 |= fdist2	        |update fdist1 with counts from fdist2|
|fdist1 < fdist2	        |test if samples in fdist1 occur less frequently than in fdist2|

<span style="font-size:2.5em; line-height:36px"><strong>4. Back to Python: Making Decisions and Taking Control
</strong></span>

<br><br><br>
## Python 학습에 관한 내용입니다. Python이 아직 익숙하지 않으면 한번 읽어봅시다.
<br><br><br><br><br><br><br><br>